---
title: "Supplemental Materials | Response to Shaw et al. (2025)"
author: "| Moi Exposito-Alonso\n| \n| Department of Integrative Biology, University
  of California Berkeley, California, USA\n| Howard Hughes Medical Institute, University
  of California Berkeley, California, USA\n| \n| Correspondance: moisesexpositoalonso@gmail.com\n"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  pdf_document:
    toc: yes
    toc_depth: 3
    latex_engine: xelatex
    fig_caption: yes
fontsize: 10pt
mainfont: Times New Roman
indent: no
header-includes:
  - \usepackage{etoolbox}
  - \usepackage{tocloft}
  - \usepackage{caption}
  - \renewcommand{\contentsname}{List of Texts}
  - \pretocmd{\caption}{\addcontentsline{toc}{section}{\protect\numberline{\thefigure}#1}}{}{}
  - \renewcommand{\figurename}{Figure S}
  - \renewcommand{\tablename}{Table S}
  - \pretocmd{\caption}{\addcontentsline{lot}{table}{\protect\numberline{\thetable}#1}}{}{}
---

\listoffigures

\listoftables

\clearpage

```{r load libraries, echo=FALSE, message=FALSE, warning=FALSE, errors=FALSE, paged.print=FALSE}
mypath<-"~/extinction/Response-GenDivMetaAnalysis/"
setwd(mypath)
knitr::opts_knit$set(root.dir = mypath)


# Dadta
library(dplyr)
library(tidyr)

# Tables
library(dplyr)
library(knitr)
library(kableExtra)
require(stringr)


# Plots
library(latex2exp)
library(scales)
library(ggplot2)
library(RColorBrewer)
library(ggside)
library(cowplot)

library(ggplot2)

library(showtext)

#Texts
showtext_auto()
font_add(family = "Times New Roman", regular=
           "/System/Library/Fonts/Supplemental/Times New Roman.ttf")
theme_set(theme_minimal(base_family = "Times New Roman"))
# Load Times New Roman (must be installed on your system)
font_add("TimesNewRoman", regular = "Times New Roman.ttf", italic = "Times New Roman Italic.ttf")
showtext_auto()

SAVEPLOTS=F

source("Response-functions.R")

```

```{r dataset, echo=F, message=FALSE, error=F, warning=F}
d<-read.csv("Supporting Data 3 (systematic review dataset)- Final Dataset.csv") %>%
  # Corrections of mistakes found
  dplyr::mutate(Common.Name = ifelse(Common.Name=="maize", "Maize", Common.Name)) # Maize appears two times, one capitalized another not

# Make some numbers numeric
d$Early.Sample.N<-as.numeric(d$Early.Sample.N)
d$Recent.Sample.N<-as.numeric(d$Recent.Sample.N)
d$Gen.interval<-as.numeric(d$Gen.interval)
d$Interval<-as.numeric(d$Interval)
d$N.Loci<-as.numeric(d$N.Loci)


# Add observation number
d$observation<-1:nrow(d)
# Generate median gen interval
d$M.Gen.interval = d$Gen.interval - median(d$Gen.interval, na.rm=T)
d$M.Year.Midpoint = d$Year.Midpoint - median(d$Year.Midpoint, na.rm=T)

write.csv(file = "tmp/d.csv",x=d)
```

```{r subset metrics, echo=F,message=FALSE}

dfm<-
  d %>%
  dplyr::filter(Diversity.Type=="AR" | Diversity.Type=="A" ) %>%
  mutate(diff=Recent.Diversity-Early.Diversity,
         diffperc=(Recent.Diversity-Early.Diversity)/Early.Diversity) %>%
  dplyr::filter(Statistical.Method==2) %>% # filter to only two time points
  dplyr::filter(is.na(Domestic.Pest.Pathogen.Status)) %>% 
  dplyr::filter(Genome == "Nuclear") %>%
  dplyr::filter(is.na(Extreme.Value))
# nrow(dfm)

dfpi<-
  d %>%
  dplyr::filter(Diversity.Type=="pi") %>%
  mutate(diff=Recent.Diversity-Early.Diversity,
         diffperc=(Recent.Diversity-Early.Diversity)/Early.Diversity) %>%
  dplyr::filter(Statistical.Method==2) %>%
  dplyr::filter(is.na(Domestic.Pest.Pathogen.Status)) %>% 
  dplyr::filter(Genome == "Nuclear") %>%
  dplyr::filter(is.na(Extreme.Value))
# nrow(pi)

dfhe<-
  d %>%
  dplyr::filter(Diversity.Type=="He") %>%
  mutate(diff=Recent.Diversity-Early.Diversity,
         diffperc=(Recent.Diversity-Early.Diversity)/Early.Diversity) %>%
  dplyr::filter(Error.Type%in% c("SD","SE")) %>%
  dplyr::filter(Statistical.Method==2) %>%
  dplyr::filter(is.na(Domestic.Pest.Pathogen.Status)) %>% 
  dplyr::filter(Genome == "Nuclear") %>%
  dplyr::filter(is.na(Extreme.Value))
# nrow(dfhe)
dfpi<-rbind(dfpi, dfhe)
# nrow(dfpi)


dmoi<-
  rbind(dfpi,dfm)

```

```{r compute_CIs, echo=F,warning=FALSE}
# Compute confidence intervals and p-values

# PI DIVERSITY
dfpici<-compute_diversity_change(dfpi)
# dfpici$pvalue<-compute_p_values_lowtest(dfpi$HedgesG, dfpi$HedgesG.err)$p_value
dfpici$pvalue<-compute_p_values(dfpi$HedgesG, dfpi$HedgesG.err)$p_value
dfpici$pvalue_signed<-dfpici$pvalue * dfpi$HedgesG/abs(dfpi$HedgesG)
dfpici$pvalue_color<- dfpici$Percent.Change/abs(dfpici$Percent.Change) *as.numeric(dfpici$pvalue<0.05/nrow(dfpici))
  dfpici$pvalue_color[is.na(dfpici$pvalue_color)]<-0
# truncate
dfpici$Percent.Change.truncated<-ifelse(dfpici$Percent.Change< (-100), -100, dfpici$Percent.Change)
dfpici$Percent.Change.truncated<-ifelse(dfpici$Percent.Change.truncated > 100, 100, dfpici$Percent.Change.truncated)
dfpici$n0 <-dfpi$Early.Sample.N
dfpici$n1 <-dfpi$Recent.Sample.N
# Add base pairs
dfpici$totalbp<-ifelse(!is.na(dfpi$N.Loci), dfpi$N.Loci,dfpi$N.bp)
# conservative p-value
dfpici$pvalueconservative<-compute_p_values_conservative(dfpi$HedgesG, dfpi$HedgesG.err,dfpi$Early.Sample.N,dfpi$Recent.Sample.N)$p_value
# Add back columns
dfpici$HedgesG<-dfpi$HedgesG
dfpici$HedgesG.err<-dfpi$HedgesG.err
dfpici$Gen.interval<-dfpi$Gen.interval
dfpici$N.Loci <-dfpi$N.Loci
dfpici$Early.Sample.N <-dfpi$Early.Sample.N
dfpici$Recent.Sample.N <-dfpi$Recent.Sample.N
dfpici$Binomen.rotl <- dfpi$Binomen.rotl
dfpici$PaperID <- dfpi$PaperID
dfpici$Z.Year.Midpoint <- dfpi$Z.Year.Midpoint
dfpici$Z.Gen.interval <- dfpi$Z.Gen.interval
dfpici$Interval <- dfpi$Interval
dfpici$Marker.Type<-dfpi$Marker.Type
dfpici$observation <- dfpi$observation
dfpici$Kingdom <- dfpi$Kingdom
dfpici$Order <- dfpi$Order
dfpici$Family <- dfpi$Family
dfpici$M.Gen.interval <- dfpi$Gen.interval-median(dfpi$Gen.interval)
dfpici$M.Year.Midpoint <- dfpi$Year.Midpoint-median(dfpi$Year.Midpoint)


############################
# ALLELIC RICHNESS
dfmci<-compute_diversity_change(dfm)
dfmci$pvalue<-compute_p_values(g_values = dfm$HedgesG, g_se = dfm$HedgesG.err)$p_value
dfmci$pvalue_signed<-dfmci$pvalue * dfm$HedgesG/abs(dfm$HedgesG)
dfmci$pvalue_color<- dfmci$Percent.Change/abs(dfmci$Percent.Change) * as.numeric(dfmci$pvalue<0.05/nrow(dfmci))
  dfmci$pvalue_color[is.na(dfmci$pvalue_color)]<-0
# truncate
dfmci$Percent.Change.truncated<-ifelse(dfmci$Percent.Change< (-100), -100, dfmci$Percent.Change)
dfmci$Percent.Change.truncated<-ifelse(dfmci$Percent.Change.truncated > 100, 100, dfmci$Percent.Change.truncated)
dfmci$n0 <-dfm$Early.Sample.N
dfmci$n1 <-dfm$Recent.Sample.N
# Add base pairs
dfmci$totalbp<-ifelse(!is.na(dfm$N.Loci), dfm$N.Loci,dfm$N.bp)
# conservative p-value
dfmci$pvalueconservative<-compute_p_values_conservative(dfm$HedgesG, dfm$HedgesG.err,dfm$Early.Sample.N,dfm$Recent.Sample.N)$p_value
# Add back columns
dfmci$HedgesG<-dfm$HedgesG
dfmci$HedgesG.err<-dfm$HedgesG.err
dfmci$Gen.interval<-dfm$Gen.interval
dfmci$N.Loci <-dfm$N.Loci
dfmci$Early.Sample.N <-dfm$Early.Sample.N
dfmci$Recent.Sample.N <-dfm$Recent.Sample.N
dfmci$Binomen.rotl <- dfm$Binomen.rotl
dfmci$PaperID <- dfm$PaperID
dfmci$Z.Year.Midpoint <- dfm$Z.Year.Midpoint
dfmci$Z.Gen.interval <- dfm$Z.Gen.interval
dfmci$Interval <- dfm$Interval
dfmci$Marker.Type<-dfm$Marker.Type
dfmci$observation <- dfm$observation
dfmci$Kingdom <- dfm$Kingdom
dfmci$Order <- dfm$Order
dfmci$Family <- dfm$Family
dfmci$M.Gen.interval <- dfm$Gen.interval-median(dfm$Gen.interval)
dfmci$M.Year.Midpoint <- dfm$Year.Midpoint-median(dfm$Year.Midpoint)


#####*********************************************************************######
# also the complete two sets
dmoip<-compute_diversity_change(dmoi)
dmoip$pvalue<-compute_p_values(dmoi$HedgesG, dmoi$HedgesG.err)$p_value
dmoip$pvalue_color<- dmoip$Percent.Change/abs(dmoip$Percent.Change) * as.numeric(dmoip$pvalue<0.05/nrow(dmoip))
dmoip$HedgesG<-dmoi$HedgesG
dmoip$HedgesG.err<-dmoi$HedgesG.err
dmoi<-
  dmoi %>%
  mutate(Diversity.Metric = ifelse(Diversity.Type %in% c("A","AR"), "M", "π" ))
dmoip$Diversity.Metric = dmoi$Diversity.Metric


# Also the original
dp<-compute_diversity_change(d)
dp$pvalue<-compute_p_values(g_values = d$HedgesG, g_se = d$HedgesG.err)$p_value
dp$Diversity.Type <-d$Diversity.Type
dp$Early.Sample.N <-d$Early.Sample.N
dp$Recent.Sample.N <-d$Recent.Sample.N
dp$N.Loci <-d$N.Loci

#####*********************************************************************######
# BACKUPS
dfpici_backup<-dfpici
dfpi_backup<-dfpi
dfmci_backup<-dfmci
dfm_backup<-dfm


# write.csv(file = "tmp/dfpici.csv",dfpici_backup)
write.csv(file = "tmp/dfpici.csv",dfpici_backup)
write.csv(file = "tmp/dfmci.csv",dfmci_backup)

```

```{r BIG_SUBSET, eval=T, echo=FALSE,warning=FALSE}

# subset
# dim(dfpici_backup)
dfpicisubset<-dfpici_backup[dfpici_backup$SE.Percent.Change<5,]
dfpisubset<-dfpi_backup[dfpici_backup$SE.Percent.Change<5,]
# dim(dfpisubset)
# write.csv(file = "tmp/dfpicisubset.csv",dfpicisubset)

# subset
# dim(dfmci_backup)
dfmcisubset<-dfmci_backup[dfmci_backup$SE.Percent.Change<5,]
dfmsubset<-dfm[dfmci_backup$SE.Percent.Change<5,]
# dim(dfmsubset)

# write.csv(file = "tmp/dfpicisubset.csv",dfpicisubset)
```

```{r BIG_SUBSET_check, eval=F, echo=F}

exageratedchange<- dfmcisubset$Percent.Change[dfmcisubset$pvalue_color != 1] %>% mean(na.rm=T)
exageratedchangemeta<- metaztest(dfmcisubset$Percent.Change[dfmcisubset$pvalue_color != 1], dfmcisubset$SE.Percent.Change[dfmcisubset$pvalue_color != 1])
exagerateddecline<- dfmcisubset$Percent.Change[dfmcisubset$pvalue_color == -1] %>% mean(na.rm=T)


table(as.numeric(dfpici$pvalue_color == -1),
      !(dfpi$Red.List.status %in% c("Not","NT", "LC"))
      ) %>% fisher.test()
table(as.numeric(dfpici$pvalue_color == 1),
      !(dfpi$Red.List.status %in% c("Not","NT", "LC"))
      ) %>% fisher.test()
table(dfpisubset$Red.List.status %in% c("Not","NT", "LC"))

table(as.numeric(dfpici$pvalue_color == 1),
      (dfpi$Gen.interval >30)
      ) %>% fisher.test()
table(as.numeric(dfpici$pvalue_color == -1),
      (dfpi$Gen.interval >30)
      ) %>% fisher.test()

```

## Text S1: Subset dataset to robust and comparable genetic diversity metrics (π and S).

### Summary of numbers from the original dataset

Shaw et al (2025) include a number of metrics of genetic composition in their analyses. The definition (from Authors) is below:\
AR (allelic richness)\
A (mean alleles)\
TA (total alleles)\
pA (total private alleles)\
NPL (number of polymorphic loci)\
Other1 (other metric associated with variant counts)\
He (expected heterozygosity)\
pi (nucleotide diversity)\
h (haplotype diversity)\
H (Shannon diversity index)\
PIC (polymorphic information content)\
NEA (number of effective alleles)\
Fis (inbreeding coefficient)\
R (mean relatedness/kinship among individuals)\
apFst (among-population fixation index)\
Freq (frequency of an allele of interest)\
NPD (mean individual nucleotide p-distance)\
BS (bandsharing score)\
Other2 (other metric associated with Variant frequencies)\
Ho (observed heterozygosity)\
SH (standardised multilocus heterozygosity)\
Ai (mean alleles per individual)\
F (inbreeding coefficient at the individual level)\
NH (number of polymorphic loci per individual)\
Other3 (other metric associated with Individual-level diversity)\
Ne (effective population size estimated from molecular data)\
Nd (effective population size estimated from demographic data)\
Nb (effective number of breeders),\
Nf (female effective population size),\
Nc (effective population size estimated from a population census, and calculated based on an assumption about the Ne:Nc ratio),\
Other4 (other metric associated with integrated statistics).\

These were further grouped in four categories (definitions from Authors):\
Group1 = metrics based on variant counts (AR, A, TA, pA, NPL, Other1)\
Group2 = metrics based on evenness of variant frequencies (He, pi, h, H, PIC, NEA, Fis, R, apFst, Freq, NPD, BS, Other2)\
Group3 = metrics based on population means of individual-level diversity (Ho, SH, Ai, F, NH, Other3)\
Group4 = integrated statistics (Ne, Nd, Nb, Nf, Nc, Other4).\

In this reanalyses, the dataset is filtered to only standard metrics of diversity:

```{r metric_count, echo=F}
d$Diversity.Type %>%
  table() %>%
  as.data.frame() %>%
  setNames(c("Diversity type", "# observations")) %>%
  arrange(-`# observations`) %>% 
  #t() %>%
  kable(.,caption="Number of observations by diversity type")


# # Prepare your table
# tbl <- d$Diversity.Type %>%
#   table() %>%
#   as.data.frame() %>%
#   arrange(desc(Freq)) %>%
#   t()
# 
# # First half (all but last 10 columns)
# kable(tbl[, 1:(ncol(tbl) - 10)],
#       caption = "Number of observations by diversity type") %>%
#   kable_styling(full_width = FALSE)
# 
# # Second half (last 10 columns only, no new caption)
# kable(tbl[, (ncol(tbl) - 9):ncol(tbl)],
#       col.names = colnames(tbl)[(ncol(tbl) - 9):ncol(tbl)]) %>%
#   kable_styling(full_width = FALSE)

```

-   We then focus on the one hand on average difference metrics, π diversity, He and π diversity. There are a total of `r sum(d$Diversity.Type=="He")` and `r sum(d$Diversity.Type=="pi")` studies with He and Pi measurements. We decided to combine them so there is sufficient sample size for downstream analyses (for biallelic loci, expected heterozygosity and π are the same value). We also focus on the number of alleles, `r sum(d$Diversity.Type=="A"|d$Diversity.Type=="AR")`, for which we also have population genetic expectations. Together, these are `r 100*sum(sum(d$Diversity.Type=="He"|d$Diversity.Type=="pi"),sum(d$Diversity.Type=="A"|d$Diversity.Type=="AR"))/nrow(d)`% of original dataset.

-   We also focus on studies with metrics in 2 time points, rather than linear trajectories or coalescent, which makes comparisons most straightforward. The number of 2 timepoint measurements is: `r table(d$Statistical.Method)["2"]`, linear: `r table(d$Statistical.Method)["1"]`, and coalescent: `r table(d$Statistical.Method)["3"]`. Coalescent is `r 100*table(d$Statistical.Method)["3"]/nrow(d)`% of the original dataset.

```{r summary_generations, echo=F,message=F}
tmp=summary(dfpi$Gen.interval)
mean_gen=format(summary(dfpi$Gen.interval)[4], digits=3)
median_gen=format(summary(dfpi$Gen.interval)[3], digits=3)
q1_gen=format(summary(dfpi$Gen.interval)[2], digits=3)
q3_gen=format(summary(dfpi$Gen.interval)[5], digits=3)


tmp=summary(dfpi$Gen.interval)
mean_gen_m=format(summary(dfpi$Gen.interval)[4], digits=3)
median_gen_m=format(summary(dfpi$Gen.interval)[3], digits=3)
q1_gen_m=format(summary(dfpi$Gen.interval)[2], digits=3)
q3_gen_m=format(summary(dfpi$Gen.interval)[5], digits=3)

```

-   The number of observations by Kingdom and Phylum mostly show observations include insects, vertebrates, and flowering plants. We do not do any filtering.

```{r check_domesticated_species_enrichment,echo=F, eval=T, message=F}
domesticated=table(is.na(d$Domestic.Pest.Pathogen.Status))["FALSE"]
message(domesticated, "observations of domesticated species")

domesticatedorpests=table(!is.na(d$Domestic.Pest.Pathogen.Status))
```

-   Number of observations of domesticated species:\
    Domesticated `r domesticated` observations (% `r 100*domesticated/nrow(d)`). Domesticated/pathogens/pests `r domesticatedorpests` observations (% `r 100*domesticatedorpests/nrow(d)`). We removed domesticated species.

```{r check_nuclear,echo=F, eval=T, message=F}
nuclear=table(d$Genome)["Nuclear"]
```

-   Number of nuclear vs other genomes:\
    Nuclear genome markers `r nuclear` observations (% `r 100*nuclear/nrow(d)`). We focus on nuclear, as chloroplast and mitochondria undergo different evolutionary dynamics (maternal inheritanze, no recombination during meiosis). There are `r table(d$Genome)["Nuclear"]` nuclear, `r table(d$Genome)["Mitochondrial"]` mitochondrial, and `r table(d$Genome)["Mixed"]` mixed genomes. Non-nuclear is `r 100-100*table(d$Genome)["Nuclear"]/nrow(d)`% of original dataset.

### Summary of numbers from filtered dataset

Total number `r nrow(dfpi)+nrow(dfm)` of which `r clean(100*nrow(dfpi)/(nrow(dfpi)+nrow(dfm)))`% are $\pi$. The final subset is `r clean(100*(nrow(dfpi)+nrow(dfm))/nrow(d))`% of the original dataset.

Summaries of allelic richness $\pi$

-- `r nrow(dfpi)` studies\
-- `r length(unique(dfpi$Binomen.rotl))` species\
-- Summary of number of generations in the final dataset is: Mean=`r printmean(dfpi$Gen.interval)` generations, Median=`r printmedian(dfpi$Gen.interval)` generations, IQR= `r printIQR(dfpi$Gen.interval)` generations.\

Summaries of allelic richness $M$\
-- `r nrow(dfm)` studies\
-- `r length(unique(dfm$Binomen.rotl))` species\
-- Summary of number of generations in the final dataset is: Mean=`r printmean(dfm$Gen.interval)` generations, Median=`r printmedian(dfm$Gen.interval)` generations, IQR= `r printIQR(dfm$Gen.interval)` generations.\

```{r fig_check_funnel, echo=F, eval=TRUE, warning=F,, fig.height=5,fig.width=12, fig.cap="Funnel plot of effect size vs error shows a classic relationship of smaller effect size in measurements with less error. \\label{fig:fig_check_funnel}" }

dmoip<-
  dmoip %>%
  mutate(Percent.Change.trunc= ifelse(Percent.Change > 100, 100, Percent.Change)) %>%
  mutate(Percent.Change.trunc= ifelse(Percent.Change.trunc < -100, -100, Percent.Change.trunc))


p1<-
ggplot(dmoip)+
  geom_point(aes(y=1/HedgesG.err,x=HedgesG, color=Diversity.Metric),shape=16,alpha=0.5)+
  scale_y_log10()+
  theme(legend.position = "none")+
  # scale_color_brewer(type = "qual")+\
  ylab("Hedgest g error inverse (1/SE effect size)")+
  xlab("Hedgest g (effect size)")+
  scale_color_manual(values = c("M"="#8B008B" ,"π"="#468B00" ) )+
  theme_minimal(base_family = "Times New Roman")

p2<-
ggplot(dmoip)+
  geom_point(aes(y=1/SE.Percent.Change,x=Percent.Change.trunc,
                 color=Diversity.Metric),shape=16,alpha=0.5)+
  # scale_y_log10()+
  # scale_y_continuous(
  #   breaks = 10^(1:6),
  #   limits = c(-6, -1),
  #   labels = TeX(sprintf("$10^{%d}$", abs(-6:-1)))
  # )+
  theme(legend.position = "none")+
  scale_color_manual(values = c("M"="#8B008B" ,"π"="#468B00" ) )+
  # xlim(c(-100,100))+
  xlab("Genetic diversity change (%)")+
  ylab("Genetic diversity change error (1/SE%)")+
  theme_minimal(base_family = "Times New Roman")

plot_grid(p1,p2)
```

Figure S \ref{fig:fig_check_funnel} shows the typical funnel shape whereby most precise estimates (high 1/SE) show smaller effects.

```{r colores, eval=F, echo=FALSE }
 "#00008B" "#46008B" "#8B008B" "#8B0046" "#8B0000" "#8B4500" "#8B8B00"
 "#468B00" "#008B00" "#008B45" "#008B8B" "#00468B"

```

### Re-calculate effect sizes

To reproduce Shaw calculation of effect size:

$$g = J \cdot \frac{\pi_1 - \pi_0}{s_{\text{g-pooled}}}$$

To calculate the corrected version we have the degrees of freedom used for effect size correction is: $df = n_{\text{early}} + n_{\text{recent}} - 2$. We used Hedges' correction factor $J$ for small-sample bias: $J = 1 - \frac{3}{4 \cdot df - 1}$.

Its standard error was estimated as:

$$SE_g = \sqrt{\frac{n_{\text{early}} + n_{\text{recent}}}{n_{\text{early}} \cdot n_{\text{recent}}}+ \frac{g^2}{2 \cdot (n_{\text{early}} +n_{\text{recent}})}}$$

### Calculate a percentage diversity change

#### Percentage change

The fraction (or percentage if multiplied $\times$ 100) of diversity change is defined as: $$X_\pi =  \frac{\pi_{present} -\pi_{past}}{\pi_{past}}$$

Each estimate came with sample size and some form of error extracted during the review process for both past (early) and present (recent) timepoints. The errors reported came in several forms: standard deviation (SD), standard error (SE), or the width of a 95% confidence interval (CI). All error types were converted to standard errors using the following rules (diversity $\pi$ and $M$ had mostly errors reported as SD or SE, only a few 95%CI and only 6 observations of $M$ with IQR, discarded):

$$\text{SE} =\begin{cases}\displaystyle \frac{\text{SD}}{\sqrt{n}} & \text{if error type is SD} \\\text{SE} & \text{if already reported as SE} \\\displaystyle \frac{\text{CI width} / 2}{1.96} & \text{if reported as 95\% CI}\end{cases}$$

where $n$ is the sample size. From this standard deviation (SD) was recomputed for completeness, as sample sizes were known for all observations as $\text{SD} = \text{SE} \times \sqrt{n}$. This was done separately for early and recent time points. Then pooled standard deviation was computed as:

$$SD_{\text{pooled}} = \sqrt{\frac{(n_{\text{early}} - 1) \cdot SD_{\text{early}}^2 + (n_{\text{recent}} - 1) \cdot SD_{\text{recent}}^2}{n_{\text{early}} + n_{\text{recent}} - 2}}$$

Just to have something informative of both studies, the effective sample size was computed as the harmonic mean of early and recent sample sizes: $n_{\text{eff}} = \frac{2 \cdot n_{\text{early}} \cdot n_{\text{recent}}}{n_{\text{early}} + n_{\text{recent}}}$.

From this, we aim to compute a error of the measured change ($X_\pi = 100 \times \frac{\pi_{present} -\pi_{past}}{\pi_{past}} = \frac{\Delta_\pi}{\pi_{past}}$) The standard error of the change is computed assuming errors in the two time points are independent and estimated the standard error of the difference: $\text{SE}_{\Delta} = \sqrt{SE_{\text{early}}^2 + SE_{\text{recent}}^2}$, and scaled to be percentage change error:$\text{SE}_{\%\Delta} = \left( \frac{\text{SE}_{\Delta}}{\pi_0} \right) \times 100$.

This standard error is not exact because of the division of $\pi_0$, which itself contains noise. We can attempt to create a second standard error uses the delta method. While both are approximations, the delta method attempts to provide a more sophisticated approximation. Consider $f(X, Y) = \left( \frac{X}{Y} - 1 \right) \times 100$, where $X = \pi$ and $Y = \pi_0$. We apply the **delta method** to approximate the variance of $f(X, Y)$. The first-order Taylor expansion is:

$$\text{Var}[f(X, Y)] \approx 
\left( \frac{\partial f}{\partial X} \right)^2 \cdot \text{Var}[X] +
\left( \frac{\partial f}{\partial Y} \right)^2 \cdot \text{Var}[Y] +
2 \cdot \frac{\partial f}{\partial X} \cdot \frac{\partial f}{\partial Y} \cdot \text{Cov}[X, Y]$$

Assuming independence between $X$ and $Y$ (i.e., $\text{Cov}[X, Y] = 0$), the partial derivatives are:

$$
\frac{\partial f}{\partial X} = \frac{100}{Y}, \quad
\frac{\partial f}{\partial Y} = -\frac{100 X}{Y^2}
$$

Substituting into the variance expression:

$$
\text{Var}[\%\Delta \pi] \approx 
\left( \frac{100}{Y} \right)^2 \cdot \text{Var}[X] +
\left( \frac{100 X}{Y^2} \right)^2 \cdot \text{Var}[Y]
$$

Thus, the **standard error** of the percent change is:

$$
\text{SE}[\%\Delta \pi] \approx \sqrt{
\left( \frac{100}{Y} \right)^2 \cdot \text{SE}_X^2 +
\left( \frac{100 X}{Y^2} \right)^2 \cdot \text{SE}_Y^2
}
$$

#### Log ratio change

Although likely an arithmetic percentage change is the most understandable metric, for completeness, I also calculate the ratio of change:

$$LR_\pi =  \text{log} \big(\frac{\pi_{present}}{\pi_{past}}\big)$$

When averaging the log ratio we get a geometric (rather than arithmetic) mean, which may be helpful.

We can also calculate the standard error of the log ratio using the delta method. Let: $X = \pi$ (recent diversity) and $Y = \pi_0$ (early diversity), and $f(X, Y) = \log\left( \frac{X}{Y} \right) = \log(X) - \log(Y)$. We aim to compute the standard error of the log-ratio, $\text{SE}\left[ \log\left( \frac{X}{Y} \right) \right]$, using the **delta method**, assuming that $X$ and $Y$ are independent.

The variance of $f(X, Y)$ is approximated as:

$$\text{Var}[f(X, Y)] \approx \left( \frac{\partial f}{\partial X} \right)^2 \cdot \text{Var}[X] +\left( \frac{\partial f}{\partial Y} \right)^2 \cdot \text{Var}[Y]$$

The partial derivatives are:

$$\frac{\partial f}{\partial X} = \frac{1}{X}, \quad\frac{\partial f}{\partial Y} = -\frac{1}{Y}$$

Substituting into the variance approximation:

$$\text{Var}\left[ \log\left( \frac{X}{Y} \right) \right] \approx\left( \frac{1}{X} \right)^2 \cdot \text{SE}_X^2 +\left( \frac{1}{Y} \right)^2 \cdot \text{SE}_Y^2$$

Therefore, the standard error of the log-ratio is:

$$\text{SE}\left[ \log\left( \frac{X}{Y} \right) \right] =\sqrt{\left( \frac{\text{SE}_X}{X} \right)^2 +\left( \frac{\text{SE}_Y}{Y} \right)^2}$$.

### Summary statistics of changes

Summaries of $\pi$ change:\
Median [IQR] Hedges g = `r printmedian(dfpi$HedgesG)` [ `r printIQR(dfpi$HedgesG)`]\
Median [IQR] Error of Hedges g = `r printmedian(dfpi$HedgesG.err)` [ `r printIQR(dfpi$HedgesG.err)`]\
Median [IQR] Percentage change = `r printmedian(dfpici$Percent.Change)`% [`r printIQR(dfpici$Percent.Change)`%]\
Median [IQR] SE of Percentage change = `r printmedian(dfpici$SE.Percent.Change)`% [`r printIQR(dfpici$SE.Percent.Change)`%]\
Mean Percentage change= `r printmean(dfpici$Percent.Change)`%\
Mean SE of Percentage change= `r printmean(dfpici$SE.Percent.Change)`%\

Summaries of $M$ change:\
Median [IQR] Hedges g = `r printmedian(dfm$HedgesG)` [ `r printIQR(dfm$HedgesG)`]\
Median [IQR] Error of Hedges g = `r printmedian(dfm$HedgesG.err)` [ `r printIQR(dfm$HedgesG.err)`]\
Median [IQR] Percentage change = `r printmedian(dfmci$Percent.Change)`% [`r printIQR(dfmci$Percent.Change)`%]\
Median [IQR] SE of Percentage change = `r printmedian(dfmci$SE.Percent.Change)`% [`r printIQR(dfmci$SE.Percent.Change)`%]\
Mean Percentage change= `r printmean(dfmci$Percent.Change)`%\

```{r check_signi_variation_subset, eval=T, warning=F, echo=FALSE,fig.width=4,fig.height=6, fig.cap="Distribution of effect sizes (g) and raw genetic diversity changes (%). \\label{fig:data}"}

plot1<-
dmoip %>%
  arrange(Diversity.Metric) %>%
ggplot()+
  geom_hline(yintercept = 0, color='darkgrey', lty="dotted")+
  geom_vline(xintercept = 0, color='darkgrey', lty="dotted")+
  geom_segment(aes(
                     y=HedgesG - 1.96*HedgesG.err, yend=HedgesG +  1.96*HedgesG.err,
                     x=Percent.Change,xend=Percent.Change,
                     color=Diversity.Metric
                     ),alpha=0.25)+
  geom_segment(aes(
                     y=HedgesG ,yend=HedgesG,
                     x=Upper.CI,xend=Lower.CI,
                     color=Diversity.Metric
                     ),alpha=0.25)+
  geom_point(aes(
                 y=HedgesG,
                 x=Percent.Change,
                 color=Diversity.Metric
                 ),shape=16, alpha=0.25
             )+

  scale_x_continuous(breaks = seq(-100,+100,by=50),
                     labels = paste(seq(-100,+100,by=50),"%"))+
  coord_cartesian(xlim=c(-100,100),ylim=c(-8,8))  +
  scale_color_manual(values = c("M"="#8B008B" ,"π"="#468B00" ) )+
  xlab("Genetic diversity change (%)")+
  ylab("Hedges' g (effect size)")+
  theme_minimal(base_family = "Times New Roman")+
  theme(legend.position = "none")

plot2<-
dmoip %>%
  arrange(Diversity.Metric) %>%
ggplot()+
  geom_histogram(aes(
                 x=Percent.Change,
                 fill=Diversity.Metric
                 ),alpha=0.5,bins = 200
             )+
  scale_fill_manual(values = c("M"="#8B008B" ,"π"="#468B00" ) )+
  coord_cartesian(xlim=c(-100,100))  +
  scale_x_continuous(breaks = seq(-100,+100,by=50),
                     labels = paste(seq(-100,+100,by=50),"%"))+
  # scale_y_log10()+
    # scale_y_log10(
  #   breaks = 10^(1:6),
  #   labels = TeX(sprintf("$10^{%d}$", abs(1:6)))
  # )+
  xlab("Genetic diversity change (%)")+
  ylab("Number of studies")+
  theme_minimal(base_family = "Times New Roman")+
  theme(legend.position = "none")
# plot2

plot_grid(plot1,plot2,ncol=1)

# Save
plotonlydiversity<-
  dfpici %>%
  # arrange(Diversity.Metric) %>%
ggplot()+
  geom_histogram(aes(
                 x=Percent.Change
                 ),alpha=0.5,bins = 200
             )+
  coord_cartesian(xlim=c(-100,100))  +
  scale_x_continuous(breaks = seq(-100,+100,by=50),
                     labels = paste(seq(-100,+100,by=50),"%"))+
  scale_y_log10(
    breaks = 10^(1:6),
    labels = TeX(sprintf("$10^{%d}$", abs(1:6)))
  )+
  xlab("Genetic diversity change (%)")+
  ylab("Number of studies")+
  theme_minimal(base_family = "Times New Roman")+
  theme(legend.position = "none")

SAVEPLOTS=F

if(SAVEPLOTS) save_plot("figs/rawdata_histogram_pi.pdf",plot = plotonlydiversity,base_height = 5, base_width = 5)
  
```

```{r top_hits, warning=FALSE, echo=F}

declinetablepi<-
table(as.numeric(dfpici$HedgesG < quantile(dfpici$HedgesG, probs = 0.05)),
      dfpici$Percent.Change < quantile(dfpici$Percent.Change, probs = 0.05)
      )

declinetablem<-
table(as.numeric(dfm$HedgesG < quantile(dfm$HedgesG, probs = 0.05)),
      dfmci$Percent.Change < quantile(dfmci$Percent.Change, probs = 0.05)
      )

```

Hedges' g and genetic diversity change in percentage are correlated since raw data is used to calculate g (Figure S\ref{fig:data}), although not perfectly (Pearson's r=`r clean(cor.test(dmoip$HedgesG,dmoip$Percent.Change)$estimate)`). The extremest values in $\pi$, which comprises the majority of the data however, typically do not coincide. We can see that of the 5% top $\pi$ values that decline based on Hedge's g (`r sum(declinetablepi[2,])`), only a small fraction are also among the top raw % values of decline (`r clean(100*declinetablepi[2,2]/sum(declinetablepi[2,]))`%). Same for $M$ (`r clean(100*declinetablem[2,2]/sum(declinetablem[2,]))`%). Figure S\ref{fig:raw} shows the raw average and estimated effect sizes in the comparision of Hedges' g and raw values of change (colors indicate significance, see section below).

\clearpage

## Text S2: Test the temporal shift of genetic diversity changes

### Calculate average change with different methods

```{r mcmctests, echo=F, warning=F, eval=F}

subsetmodel<-metamcmc(dfpicisubset, variable="Percent.Change", se="SE.Percent.Change")

# paste(clean(HPDinterval(subsetmodel$Sol[, "(Intercept)"])),collapse=' - ')
clean(mean(subsetmodel$Sol[, "(Intercept)"]))
dfpicisubset$Percent.Change %>% mean(na.rm=T)

posteriormcmc(subsetmodel)

```

```{r meta_hedge, echo=F,warning=F,message=FALSE}
# Classic inverse variance test
metazg<-metaztest(dfpici$HedgesG, dfpici$HedgesG.err )
metazperc<-metaztest(dfpici$Percent.Change, dfpici$SE.Percent.Change)

metazgall<-metaztest(d$HedgesG, d$HedgesG.err )


# MCMCglmm based metaregression
library(MCMCglmm)
rerun=TRUE
if(rerun){
  mcmcgall<-metamcmc(d, variable="HedgesG", se="HedgesG.err")

  mcmcg<-metamcmc(dfpici, variable="HedgesG", se="HedgesG.err")
                     # nitt = 6000000, burnin = 200000, thin = 5000)
  mcmcperc<-metamcmc(dfpici, variable="Percent.Change", se="SE.Percent.Change")
                     # nitt = 6000000, burnin = 200000, thin = 5000)
  # mcmcperconlydecline<-metamcmc(dfpici[dfpici$pvalue_color == -1, ], variable="Percent.Change", se="SE.Percent.Change")

  save(file = "tmp/mcmcgall.rda", mcmcgall)
  save(file = "tmp/mcmcg.rda", mcmcg)
  save(file = "tmp/mcmcperc.rda", mcmcperc)
}else{
  load(file = "tmp/mcmcgall.rda")
  load(file = "tmp/mcmcg.rda")
  load(file = "tmp/mcmcperc.rda")
}

```

```{r meta_m_hedge, echo=F,warning=F,message=FALSE}
# Classic inverse variance test
mmetazg<-metaztest(dfmci$HedgesG, dfmci$HedgesG.err )
mmetazperc<-metaztest(dfmci$Percent.Change, dfmci$SE.Percent.Change)


# MCMCglmm based metaregression
library(MCMCglmm)
rerun=TRUE
if(rerun){
  mmcmcg<-metamcmc(dfmci, variable="HedgesG", se="HedgesG.err")
                     # nitt = 6000000, burnin = 200000, thin = 5000)
  mmcmcperc<-metamcmc(dfmci, variable="Percent.Change", se="SE.Percent.Change")
                     # nitt = 6000000, burnin = 200000, thin = 5000)
  save(file = "tmp/mmcmcg.rda", mmcmcg)
  save(file = "tmp/mmcmcperc.rda", mmcmcperc)
}else{
  load(file = "tmp/mmcmcg.rda")
  load(file = "tmp/mmcmcperc.rda")
}

```

*For diversity* $\pi$\

Arithmetic mean Hedge's g = `r printmean(dfpici$HedgesG)` ±`r printCI(dfpici$HedgesG)` (95% CI).\
Weighted SE mean Hedge's g = `r clean(metazg$Weighted_mean)` ± `r clean(metazg$CI_half_width)`(95% CI) (z = `r clean(metazg$z)`, $P$ = `r metazg$p`).\
MCMCglmm mean (intercept) of Hedge's g = `r posteriormcmc(mcmcg)$summary`.

 

Arithmetic mean % $\pi$ = `r printmean(dfpici$Percent.Change)`% ± `r printCI(dfpici$Percent.Change)`% (95% CI).\
Weighted SE mean % $\pi$ = `r clean(metazperc$Weighted_mean)`% ± `r clean(metazperc$CI_half_width)`% (95% CI) (z = `r clean(metazperc$z)`, $P$ = `r metazperc$p`).\
MCMCglmm mean (intercept) of % $\pi$ = `r posteriormcmc(mcmcperc)$summary` .

Geometric mean % $\pi$ = `r printmean((exp(mean(dfpici$Log.Change)) - 1) * 100)`% ± `r clean((exp(printCI(dfpici$Log.Change)) - 1) * 100)`% (95% CI).\
Weighted SE geometric mean % $\pi$ = `r clean((exp(mean(metaztest(dfpici$Log.Change, dfpici$SE.Log.Change)$Weighted_mean)) - 1) * 100)`% ± `r clean((exp(mean(metaztest(dfpici$Log.Change, dfpici$SE.Log.Change)$CI_half_width)) - 1) * 100)`% (95% CI).\

*For allelic richness* $M$\

Average Hedge's g = `r printmean(dfmci$HedgesG)` ±`r printCI(dfmci$HedgesG)` (95% CI).\
Weighted SE mean Hedges'g = `r clean(mmetazg$Weighted_mean)` ± `r clean(mmetazg$CI_half_width)`(95% CI) (z = `r clean(mmetazg$z)`, $P$ = `r mmetazg$p`).\
MCMCglmm mean (intercept) of Hedges'g = `r posteriormcmc(mmcmcg)$summary`.

 

Arithmetic mean % $M$ = `r printmean(dfmci$Percent.Change)`% ± `r printCI(dfmci$Percent.Change)`% (95% CI).\
Weighted SE mean % $M$ = `r clean(mmetazperc$Weighted_mean)` ± `r clean(mmetazperc$CI_half_width)`(95% CI) (z = `r clean(mmetazperc$z)`, $P$ = `r mmetazperc$p`).\
MCMCglmm mean (intercept) of % $M$ = `r posteriormcmc(mmcmcperc)$summary`.

Geometric mean % $\pi$ = `r printmean((exp(mean(dfmci$Log.Change)) - 1) * 100)`% ± `r clean((exp(printCI(dfmci$Log.Change)) - 1) * 100)`% (95% CI).\
Weighted SE geometric mean % $\pi$ = `r clean((exp(mean(metaztest(dfmci$Log.Change, dfmci$SE.Log.Change)$Weighted_mean)) - 1) * 100)`% ± `r clean((exp(mean(metaztest(dfmci$Log.Change, dfmci$SE.Log.Change)$CI_half_width)) - 1) * 100)`% (95% CI).\

### Variations of meta-regression

To study the robustness of the original MCMCglmm meta-regression from Shaw, we conduct several variations of their function. From their supplement, they run MCMCglmm( g \~ Z.year.Midpoint + Z.Gen.interval, random= \~PaperID) using weak priors. Following Shaw, I used a weak inverse-gamma model prior (V=1, nu=0.002) Aka. "pweak".

I generated variations over this model to understand robustness and explain why the results may differ from an arithmetic or error-weigthed mean (models were named numerically with increasing fixed or random terms, "m1"..."m6"):

Changing fixed effects. I first changed the mean-centered generation interval fixed effect (shaw1 model). The mean-centering is necessary so that the intercept corresponds with the average fixed effect variable. However, the generation interval was not normally distributed. In fact, \>75% of the dataset showed lower number of generations (quantile 75% = 6 generations) than the mean (6.5 generations). This is due to a long tail effect. A approach less driven by long tails is including a fixed effect of generation interval median-centered. When conducting that, the effect typically becomes much weaker or non-significant (this corrected model was named shaw2).

Changing random effects. Shaw includes Paper ID, and I tested the inclusion of Kingdom, species, and Marker type, to account for other possible heterogeneity in the data.

Changing priors. Although a weak prior seems sensible, in Bayesian statistics is important to test model robustness based on priors, especially when signals appear weak. I tested not including any prior (using default MCMCglmm, nu=0, V=1, aka. "none"). I tested a more regularizing prior "preg" (V=1e-4, nu=10), and a prior that would limit unexplained variance to focus on the observation measurement error term ($1/SE^2$), aka "pfix" (V=1e-4, nu=100). The last one is closer to fixed effect meta-analysis. These priors represent a range from low to strong shrinking, from weak information but unstable estimates to stronger information in model from standard error measurements.

The same set of models was fitted to the entire dataset (using effect sizes) as well as standard average diversity $\pi$ and allelic richness $M$ both effect sizes and percentage change. Models in reporting tables are ranked by DIC from MCMCglmm.

Several top models in both $\pi$ and $M$ percentage changes included non-significant, positive and negatively significant trends.

#### *All Shaw data effect size*

```{r mcmcrobustness_all , eval=T, echo=F}
RERUN=F

if(RERUN){
  
  source('response-mcmc-models.R')
  
}else{
  
  model_results<-read.csv(file = "tmp/metareg_model_results_hedge_alldata.csv")  %>%
    arrange(-center)
}

model_results %>%
  dplyr::mutate(significant = ifelse(significant, "*", "ns")) %>%
  dplyr::mutate(summary_string = gsub(" (95% HPI)", "", summary_string, fixed = TRUE)) %>%
  dplyr::mutate(summary_string = paste(summary_string, significant)) %>%
  dplyr::mutate(DIC = round(10 * DIC) / 10) %>%
  dplyr::select(label, DIC, summary_string, fixed_str, random_str) %>%
  kable("latex", booktabs = TRUE, longtable = TRUE,
        escape = TRUE,
        caption = "Shaw effect size meta-regression model comparison.",
        col.names = c("label", "DIC", "summary", "fixed effects", "random effects"),
        align = c("l", "r", "l", "p{5cm}", "p{5cm}")) %>%
  kable_styling(latex_options = c("striped", "hold_position", "repeat_header"))

```

The exact model from Shaw et al. ("mshaw_pweak") recapitulates the found g=-0.11 reported int he paper. We note that the fixed erffect generation interval is non-significant, indicating that with increasing generation interval there is a positive trend in the effect size.

```{r shawmodel, eval=T, warning=FALSE, echo=FALSE}
load("tmp/mshaw.rda")
summary(mshaw)$solutions %>% 
  kable(caption = "Shaw meta-regression summary.",
)
```

Exclusion of the key diversity variables makes the Shaw model non-significant.

```{r shawmodelonepi, eval=T, warning=FALSE, echo=FALSE}
load("tmp/mshawnopim.rda")
summary(mshawnopim)$solutions %>% 
  kable(caption = "Shaw meta-regression summary excluding π and M.",
)

```

#### *Average genetic diversity* $\pi$ percent change

```{r mcmcrobustness_percent , eval=T, echo=F, warning=FALSE, error=FALSE, message=F}
RERUN=F

if(RERUN){
  
  source("response-mcmc-models.R")
  
}else{
  
  model_results<-read.csv(file = "tmp/metareg_model_results.csv")  %>%
    arrange(-center)
}

# parse model results
model_results <- model_results %>%
  mutate(
    prior_type = case_when(
      prior_str == "none" ~ "none",
      str_detect(prior_str, "nu=1e\\+06|nu=1e\\+06") ~ "fixed-like",
      str_detect(prior_str, "nu=0.002") ~ "weak",
      str_detect(prior_str, "nu=10") ~ "moderate",
      TRUE ~ "custom"
    )
  )

# knit the table
model_results %>%
  dplyr::mutate(significant = ifelse(significant, "*", "ns")) %>%
  dplyr::mutate(summary_string = gsub(" (95% HPI)", "", summary_string, fixed = TRUE)) %>%
  dplyr::mutate(summary_string = paste(summary_string, significant)) %>%
  dplyr::mutate(DIC = round(10 * DIC) / 10) %>%
  dplyr::select(label, DIC, summary_string, fixed_str, random_str) %>%
  # View
  # print
  kable("latex", booktabs = TRUE, longtable = TRUE,
        escape = TRUE,
        caption = "Percentage diversity π meta-regression model comparison.",
        col.names = c("label", "DIC", "summary", "fixed effects", "random effects"),
        align = c("l", "r", "l", "p{5cm}", "p{5cm}")) %>%
  kable_styling(latex_options = c("striped", "hold_position", "repeat_header"))

```

#### *Average genetic diversity* $\pi$ percent change (delta method for SE)

```{r mcmcrobustness_percent_delta , eval=T, echo=F, warning=FALSE, error=FALSE, message=F}
RERUN=F

if(RERUN){
  
  source("response-mcmc-models-deltaSE.R")
  
}else{
  
  model_results<-read.csv(file = "tmp/metareg_model_results_deltaSE.csv")  %>%
    arrange(-center)
}

# parse model results
model_results <- model_results %>%
  mutate(
    prior_type = case_when(
      prior_str == "none" ~ "none",
      str_detect(prior_str, "nu=1e\\+06|nu=1e\\+06") ~ "fixed-like",
      str_detect(prior_str, "nu=0.002") ~ "weak",
      str_detect(prior_str, "nu=10") ~ "moderate",
      TRUE ~ "custom"
    )
  )

# knit the table
model_results %>%
  dplyr::mutate(significant = ifelse(significant, "*", "ns")) %>%
  dplyr::mutate(summary_string = gsub(" (95% HPI)", "", summary_string, fixed = TRUE)) %>%
  dplyr::mutate(summary_string = paste(summary_string, significant)) %>%
  dplyr::mutate(DIC = round(10 * DIC) / 10) %>%
  dplyr::select(label, DIC, summary_string, fixed_str, random_str) %>%
  # View
  # print
  kable("latex", booktabs = TRUE, longtable = TRUE,
        escape = TRUE,
        caption = "Percentage diversity π (delta SE) meta-regression model comparison.",
        col.names = c("label", "DIC", "summary", "fixed effects", "random effects"),
        align = c("l", "r", "l", "p{5cm}", "p{5cm}")) %>%
  kable_styling(latex_options = c("striped", "hold_position", "repeat_header"))

```

#### *Average genetic diversity* $\pi$ log ratio (geometric)

```{r mcmcrobustness_percent_ratio , eval=T, echo=F, warning=FALSE, error=FALSE, message=F}
RERUN=F

if(RERUN){
  
  source("response-mcmc-models-ratio.R")
  
}else{
  
  model_results<-read.csv(file = "tmp/metareg_model_results_logratio.csv")  %>%
    arrange(-center)
}

# parse model results
model_results <- model_results %>%
  mutate(
    prior_type = case_when(
      prior_str == "none" ~ "none",
      str_detect(prior_str, "nu=1e\\+06|nu=1e\\+06") ~ "fixed-like",
      str_detect(prior_str, "nu=0.002") ~ "weak",
      str_detect(prior_str, "nu=10") ~ "moderate",
      TRUE ~ "custom"
    )
  )

# knit the table
model_results %>%
  dplyr::mutate(significant = ifelse(significant, "*", "ns")) %>%
  dplyr::mutate(summary_string = gsub(" (95% HPI)", "", summary_string, fixed = TRUE)) %>%
  dplyr::mutate(summary_string = paste(summary_string, significant)) %>%
  dplyr::mutate(DIC = round(10 * DIC) / 10) %>%
  dplyr::select(label, DIC, summary_string, fixed_str, random_str) %>%
  # View
  # print
  kable("latex", booktabs = TRUE, longtable = TRUE,
        escape = TRUE,
        caption = "Log ratio diversity π meta-regression model comparison.",
        col.names = c("label", "DIC", "summary", "fixed effects", "random effects"),
        align = c("l", "r", "l", "p{5cm}", "p{5cm}")) %>%
  kable_styling(latex_options = c("striped", "hold_position", "repeat_header"))

```

#### *Allelic richness* $M$ percent change

```{r mcmcrobustness_percent_m , eval=T, echo=F, warning=FALSE, error=FALSE, message=F}
RERUN=F

if(RERUN){
  
  source("response-mcmc-models.R")
  
}else{
  
  model_results<-read.csv(file = "tmp/metareg_model_results_m.csv")  %>%
    arrange(-center)
}

# parse model results
model_results <- model_results %>%
  mutate(
    prior_type = case_when(
      prior_str == "none" ~ "none",
      str_detect(prior_str, "nu=1e\\+06|nu=1e\\+06") ~ "fixed-like",
      str_detect(prior_str, "nu=0.002") ~ "weak",
      str_detect(prior_str, "nu=10") ~ "moderate",
      TRUE ~ "custom"
    )
  )

# knit the table
model_results %>%
  dplyr::mutate(significant = ifelse(significant, "*", "ns")) %>%
  dplyr::mutate(summary_string = gsub(" (95% HPI)", "", summary_string, fixed = TRUE)) %>%
  dplyr::mutate(summary_string = paste(summary_string, significant)) %>%
  dplyr::mutate(DIC = round(10 * DIC) / 10) %>%
  dplyr::select(label, DIC, summary_string, fixed_str, random_str) %>%
  # View
  # print
  kable("latex", booktabs = TRUE, longtable = TRUE,
        escape = TRUE,
        caption = "Percentage allelic richness M meta-regression model comparison.",
        col.names = c("label", "DIC", "summary", "fixed effects", "random effects"),
        align = c("l", "r", "l", "p{5cm}", "p{5cm}")) %>%
  kable_styling(latex_options = c("striped", "hold_position", "repeat_header"))

```

#### *Allelic richness* $M$ log ratio (geometric)

```{r mcmcrobustness_log_m , eval=T, echo=F, warning=FALSE, error=FALSE, message=F}
RERUN=F

if(RERUN){
  
  source("response-mcmc-models-ratio.R")
  
}else{
  
  model_results<-read.csv(file = "tmp/metareg_model_results_logratio_m.csv")  %>%
    arrange(-center)
}

# parse model results
model_results <- model_results %>%
  mutate(
    prior_type = case_when(
      prior_str == "none" ~ "none",
      str_detect(prior_str, "nu=1e\\+06|nu=1e\\+06") ~ "fixed-like",
      str_detect(prior_str, "nu=0.002") ~ "weak",
      str_detect(prior_str, "nu=10") ~ "moderate",
      TRUE ~ "custom"
    )
  )

# knit the table
model_results %>%
  dplyr::mutate(significant = ifelse(significant, "*", "ns")) %>%
  dplyr::mutate(summary_string = gsub(" (95% HPI)", "", summary_string, fixed = TRUE)) %>%
  dplyr::mutate(summary_string = paste(summary_string, significant)) %>%
  dplyr::mutate(DIC = round(10 * DIC) / 10) %>%
  dplyr::select(label, DIC, summary_string, fixed_str, random_str) %>%
  # View
  # print
  kable("latex", booktabs = TRUE, longtable = TRUE,
        escape = TRUE,
          caption = "Log ratio diversity M meta-regression model comparison.",
        col.names = c("label", "DIC", "summary", "fixed effects", "random effects"),
        align = c("l", "r", "l", "p{5cm}", "p{5cm}")) %>%
  kable_styling(latex_options = c("striped", "hold_position", "repeat_header"))

```


### Test per-study statistical differences: are there more or less positive trends?

Is the distribution of Hedge's g significantly different from zero for a given study? To test this, we follow a t-statistic distribution, which is the expected for Hedge's g . In addition we also conduct the test with a Gaussian distribution (which may be more akin to the meta-regression modeling, but this Normal distribution causes less significant results than t-distribution because is wider and thus more conservative). Figure S\ref{fig:significanceplot} shows all studies ranked by $-log_{10}(p)$ of Hedge's g test per study, which has a substantial inflation. Figure \ref{fig:qq} shows the inflation of P-values assuming Hedges'g follow a t-distribution, making our detection of significant effects anti-conservative.

```{r check_significance_up_low , eval=T, echo=F}
# Example counts
negative <- table(dfpici$pvalue_color)[1]
nonsignificant <- table(dfpici$pvalue_color)[2]
positive <- table(dfpici$pvalue_color)[3]

# Total significant results
total_significant <- positive + negative

# Binomial test: is probability of positive = 0.5?
gbtest<-binom.test(positive, total_significant, p = 0.5, alternative = "two.sided")


# Example counts
negative <- table(dfmci$pvalue_color)[1]
nonsignificant <- table(dfmci$pvalue_color)[2]
positive <- table(dfmci$pvalue_color)[3]

# Total significant results
total_significant <- positive + negative

# Binomial test: is probability of positive = 0.5?
mbtest<-binom.test(positive, total_significant, p = 0.5, alternative = "two.sided")

```

Distribution of positive and negative π div trajectory between two time points. Using Hedge's g for significance using two sided test and family-wise Bonferroni $P$-value correction 0.05/number of tests. The number of significant tests is as follows: Non-significant `r table(dfpici$pvalue_color)[2]`. Declining `r table(dfpici$pvalue_color)[1]`. Increasing `r table(dfpici$pvalue_color)[3]`.

-   *For diversity π*\
    Number significant trends (i.e. either positive or negative) = `r sum(abs(dfpici$pvalue_color)==1)` of total `r nrow(dfpici)` points.\
    Percentage of significant trends (i.e. either positive or negative) = `r clean(100*sum(abs(dfpici$pvalue_color)==1)/nrow(dfpici))`%\

-   *For richness M*\
    Number significant trends (i.e. either positive or negative) = `r sum(abs(dfmci$pvalue_color)==1)` of total `r nrow(dfmci)` points.\
    Percentage of significant trends (i.e. either positive or negative) = `r clean(100*sum(abs(dfmci$pvalue_color)==1)/nrow(dfmci))`%\

Are the number of significantly negative (decline of genetic diversity) and positive (increase in genetic diversity) more or less common? We conduct a Binomial test of enrichment of significant tests that are negative vs positive:

-   *For diversity π*\
    Prob. success (i.e. positive vs negative) = `r gbtest$estimate`\
    P value = `r gbtest$p.value`\
    There are also `r table(dfpici$pvalue == 0)[2]` $P$-values=0 (`r format(100 * table(dfpici$pvalue == 0)[2]/ nrow(dfpici), digits=3)`% of dataset). The cases of $P$-value=0 are roughly equal in both directions, `r (table(dfpici$pvalue_color, dfpici$pvalue==0) )[1,2]` decreasing, `r (table(dfpici$pvalue_color, dfpici$pvalue==0) )[3,2]` increasing.

-   *For richness M*\
    Prob. success (i.e. positive vs negative) = `r mbtest$estimate`\
    P value = `r mbtest$p.value`\
    There are also `r table(dfmci$pvalue == 0)[2]` $P$-values=0 (`r format(100 * table(dfmci$pvalue == 0)[2]/ nrow(dfmci), digits=3)`% of dataset). The cases of $P$-value=0 are roughly equal in both directions, `r (table(dfmci$pvalue_color, dfmci$pvalue==0) )[1,2]` decreasing, `r (table(dfmci$pvalue_color, dfmci$pvalue==0) )[3,2]` increasing.

```{r fig:qq, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, fig.height=3, fig.width=3, fig.cap="P-value distribution in QQ plot.\\label{fig:qq}"}
dfpici$pvalueconservative[is.na(dfpici$pvalueconservative)]<-1
dfpici$pvalue[is.na(dfpici$pvalue)]<-1

mypvalues<-sort(-log10(dfpici$pvalue+1e-17))
mypvaluesconservative<-sort(-log10(dfpici$pvalueconservative+1e-17))
exppvalues<-sort(runif(n = length(mypvalues)))
dataqq<-data.frame(obs=mypvalues,
                   obscons=mypvaluesconservative,
                   exp=exppvalues
                   )
ggplot(dataqq)+
  geom_point(aes(y=obs,
                 x=exp), alpha=0.5
             )+
  geom_abline()+
  xlim(c(0,10))+ylim(c(0,17))+
  xlab(TeX("Expected $-log_{10}P$") )+ylab(TeX("Observed $-log_{10}P$") )+
  theme_minimal(base_family = "Times New Roman")+
    theme(
    # panel.grid.major = element_blank(),  # remove major gridlines
    # panel.grid.minor = element_blank(),   # remove minor gridlines
    # axis.line = element_blank()  # remove major gridlines
    axis.ticks = element_line(color = "grey"),     # turn ticks back on
    axis.ticks.length = unit(0.1, "cm")
  )+
  theme(legend.position = "none")


```

```{r check_original_dataset,echo=FALSE, eval=FALSE}
# Do the original data without any filters too

# Checking that the general dataset does not show anything different
dp$Percent.Change %>% summary

# Check p-values without correction
table(dp$pvalue<0.05)

```

```{r recheck_all_ ,echo=FALSE, eval=T, message=F,warning=F}
dp$pvalue_color<- (dp$pvalue<0.05/nrow(dp))* dp$Percent.Change/abs(dp$Percent.Change)
dp$pvalue_color[is.na(dp$pvalue_color)]<-0

## Example counts
negative <- table(dp$pvalue_color)[1]
nonsignificant <- table(dp$pvalue_color)[2]
positive <- table(dp$pvalue_color)[3]

## Total significant results
total_significant <- positive + negative

## Binomial test: is probability of positive = 0.5?
gbtest<-binom.test(positive, total_significant, p = 0.5, alternative = "two.sided")


gbtestall<- binom.test(sum(d$HedgesG<0,na.rm = T), sum(!is.na(d$HedgesG)),
                       p = 0.5, alternative = "less")

#######
# # Check without family wise correction
# dp$pvalue_color<- (dp$pvalue<0.05)* dp$Percent.Change/abs(dp$Percent.Change)
# dp$pvalue_color[is.na(dp$pvalue_color)]<-0
#
# ## Example counts
# negative <- table(dp$pvalue_color)[1]
# nonsignificant <- table(dp$pvalue_color)[2]
# positive <- table(dp$pvalue_color)[3]
#
# ## Total significant results
# total_significant <- positive + negative
#
# ## Binomial test: is probability of positive = 0.5?
# gbtestnaive<-binom.test(positive, total_significant, p = 0.5, alternative = "two.sided")

```

We also test this with all the data points from Shaw et al (`r nrow(dp)`) to make sure this is not a subset effect:

-   *For all dataset*\
    Prob. success (i.e. positive vs negative) = `r gbtest$estimate`\
    P value = `r gbtest$p.value`

In addition, I compute the same binomial test without test statistics. Simply, count how many studies have Hedges' g (effect size) that is negative (`r sum(d$HedgesG<0,na.rm = T)`) out of all the dataset (`r sum(!is.na(d$HedgesG))`) and ask whether this is different from 50/50. There is no significant divergence from 50/50, with estimate `r gbtestall$estimate` and P-value=`r gbtestall$p.value`.

```{r fig:fig_change_pi, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, fig.height=3, fig.width=6, fig.cap="Relationship between raw values of diversity change π and Hedge's g significance.\\label{fig:significanceplot}"}

meanpercentage=mean(dfpici$Percent.Change[dfpici$Percent.Change > quantile(dfpici$Percent.Change,0.005) & dfpici$Percent.Change < quantile(dfpici$Percent.Change,0.995)])
metapercentage=mean(mcmcperc$Sol[, "(Intercept)"])

# View 3
myplot<-
ggplot(dfpici)+
  geom_hline(yintercept = 0, color='darkgrey')+
  geom_hline(yintercept =meanpercentage, color='#6ABD45')+
  geom_hline(yintercept =metapercentage, color='#6ABD45')+
  geom_segment(aes(y=Lower.CI,yend=Upper.CI,
                   x=-log10(pvalue+1e-17),xend=-log10(pvalue+1e-17),
                   color=factor(pvalue_color)
                   ),alpha=0.5)+
  geom_point(aes(y=Percent.Change,
                 x=-log10(pvalue+1e-17),
                 color=factor(pvalue_color)
                 ),shape=16,alpha=0.5
             )+
  scale_color_manual("Significance", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  # add side histogram
  geom_xsidehistogram(aes(x=-log10(pvalue+1e-17)), fill='darkgrey')+
  # Tweaks
  ylab("Genetic diversity change (% π)")+
  scale_y_continuous(breaks = seq(-200,+200,by=50), labels = paste(seq(-200,+200,by=50),"%"))+
  coord_cartesian(ylim=c(-200,200),)  +
  xlab(TeX("Significance ($-log_{10}$ $P$-value)"))+
  theme_minimal(base_family = "Times New Roman")+
    theme(
    panel.grid.major = element_blank(),  # remove major gridlines
    panel.grid.minor = element_blank(),   # remove minor gridlines
    # axis.line = element_blank()  # remove major gridlines
    axis.ticks = element_line(color = "grey"),     # turn ticks back on
    axis.ticks.length = unit(0.1, "cm")
  )+
  theme(legend.position = "none")
# myplot

if(SAVEPLOTS){
  ggsave(filename = "figs/rawdata_significance.pdf",
            plot = myplot+coord_cartesian(ylim=c(-200,200)) ,
            height = 6,width = 4
            )
}

plot_grid(myplot, (myplot+  coord_cartesian(ylim=c(-50,50)) ))
```

```{r fig:fig_change_pi_vs_hedge, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, fig.height=3, fig.width=6,fig.cap="Relationship between raw values of diversity change π and Hedge's g values. \\label{fig:raw}" }

myplot2<-
dfpici %>%
  arrange(abs(pvalue_color)) %>%
ggplot()+
  geom_hline(yintercept = 0, color='darkgrey')+
  geom_vline(xintercept = 0, color='darkgrey')+
  geom_segment(aes(
                     x=HedgesG - 1.96*HedgesG.err,xend=HedgesG +  1.96*HedgesG.err,
                     y=Percent.Change,yend=Percent.Change,
                     color=factor(pvalue_color)
                     ))+
  geom_segment(aes(
                     x=HedgesG ,xend=HedgesG,
                     y=Upper.CI,yend=Lower.CI,
                     color=factor(pvalue_color)
                     ))+
  geom_point(aes(
                 x=HedgesG,
                 y=Percent.Change,
               color=factor(pvalue_color)
                 ),shape=16
             )+
  scale_color_manual("Significance", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  # scale_size_continuous(range = c(0.5,3))+
  scale_y_continuous(breaks = seq(-200,+200,by=50), labels = paste(seq(-200,+200,by=50),"%"))+
  coord_cartesian(ylim=c(-200,200),xlim=c(-8,8))  +
  # xlab(TeX("Coefficient of Variation "))+
  # xlab("")+
  ylab("Genetic diversity change (%)")+
  xlab("Hedges' g (effect size)")+
  geom_vline(xintercept = -0.11, color='#6ABD45')+
  geom_hline(yintercept =meanpercentage, color='#6ABD45')+
  geom_hline(yintercept =metapercentage, color='#6ABD45')+
  theme_minimal(base_family = "Times New Roman")+
    theme(
    panel.grid.major = element_blank(),  # remove major gridlines
    panel.grid.minor = element_blank(),   # remove minor gridlines
    # axis.line = element_blank()  # remove major gridlines
    axis.ticks = element_line(color = "grey"),     # turn ticks back on
    axis.ticks.length = unit(0.1, "cm")
  )+
  theme(legend.position = "none")
 # myplot2

if(SAVEPLOTS){
  ggsave(filename = "figs/rawdata_vs_hedge.pdf",
            plot = myplot2+coord_cartesian(ylim=c(-100,100), xlim=c(-2.5, +2.5)),
            height = 4,width = 4
            )
}

plot_grid(
myplot2,
myplot2+coord_cartesian(ylim=c(-100,100), xlim=c(-2.5, +2.5))
)
```

### Test relationship with Red List

```{r check_signi_redlist, echo=F,warning=FALSE, eval=TRUE}
redlistcolors<-c(EX="black",CR="#C4362F",EN="#E58630",VU="#E2D149",NT="#00B491",LC="#008D71", DD="lightpink")

redlisttable<-table(
      as.numeric(dfpici$pvalue_color == -1),
      !(dfpi$Red.List.status %in% c("LC","DD"))
      )
redlistenrichment<-
  redlisttable %>% fisher.test

redlisttablepos<-table(
      as.numeric(dfpici$pvalue_color == 1),
      !(dfpi$Red.List.status %in% c("LC","DD"))
      )
redlistenrichmentpos<-
  redlisttablepos %>% fisher.test


```

```{r fig:redlist, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, fig.height=3, fig.width=2.5, fig.cap="Significant decline trends and Red List categories. \\label{fig:redlist}"}
redlisttable<-table(
      as.numeric(dfpici$pvalue_color == -1),
      dfpi$Red.List.status )
redlisttable<-
  data.frame(Significant= redlisttable[2,] / apply(redlisttable,2,sum),
             RedList=colnames(redlisttable)
             )

ggplot(redlisttable) +
  geom_point(
    aes(y= Significant*100,
        x = RedList, xend = RedList,
        color = RedList),
    lwd = 4, size=4
  ) +
  geom_segment(
    aes(y= Significant*100,yend=0,
        x = RedList, xend = RedList,
        color = RedList),
  ) +
  scale_color_manual(values = redlistcolors) +
  scale_x_discrete(limits = names(redlistcolors)) + # Ensure x-axis order
  # coord_flip() + # Optional: Flip the coordinates for a different layout
  theme_minimal() +
  labs(y = "Studies with significant declines (%)", x = "")+
  theme(legend.position = "none")+
  theme_minimal(base_family = "Times New Roman")#+

```

The number of significant decline cases in $\pi$ (`r sum(as.numeric(dfpici$pvalue_color == -1))`), was significantly enriched with species in the Red List (all but least concern [LC] or data deficient [DD], `r sum(!(dfpi$Red.List.status %in% c("LC","DD")))`, Fisher's test $P$=`r redlistenrichment$p.value`, Odds=`r redlistenrichment$estimate`) (Fig. \ref{fig:redlist}). The number of significant genetic diversity increase cases too (Fisher's test $P$=`r redlistenrichmentpos$p.value`, Odds=`r redlistenrichmentpos$estimate`). This was mostly driven by vulnerable species (\ref{fig:redlist}).

### Test differences with increasing time.

```{r fig:check_signi_lengthstudy, eval=T, echo=FALSE,warning=FALSE,fig.width=6,fig.height=6,fig.pos='!H' , fig.cap="Relationship between genetic changes and length of time interval. \\label{fig:timeenrich}"}

plota<-
ggplot(dfpici)+
  # geom_hline(yintercept = 0, color='darkgrey')+
  geom_segment(aes(y=dfpi$HedgesG - 1.96*dfpi$HedgesG.err,yend=dfpi$HedgesG +  1.96*dfpi$HedgesG.err,
                   # size=-log10(pvalue+1e-17),
                   x=dfpi$Interval+0.1,
                   xend=dfpi$Interval+0.1,
                   color=factor(pvalue_color)
                   ))+
  geom_point(aes(y=dfpi$HedgesG,
                 size=-log10(pvalue+1e-17),
                 x=dfpi$Interval+0.1,
                 color=factor(pvalue_color)
                 ),shape=16
             )+
  scale_x_log10()+
  scale_color_manual("Significance", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  # Tweaks
    scale_size_continuous(range = c(0.5,3))+

  ylab("Hedges' g")+
  coord_cartesian(ylim=c(-5,5))  +
  xlab(TeX("Time interval (y)"))+
  # xlab("")+
  theme_minimal(base_family = "Times New Roman")+
  # scale_x_reverse()+
  # theme(legend.position = "none", axis.text.x = element_blank())+
  theme(legend.position = "none")

plotb<-
ggplot(dfpici)+
  # geom_hline(yintercept = 0, color='darkgrey')+
  geom_segment(aes(y=Lower.CI,yend=Upper.CI,
                   # size=-log10(pvalue+1e-17),
                   x=dfpi$Interval+0.1,
                   xend=dfpi$Interval+0.1,
                   color=factor(pvalue_color)
                   ))+
  geom_point(aes(y=Percent.Change,
                 size=-log10(pvalue+1e-17),
                 x=dfpi$Interval+0.1,
                 color=factor(pvalue_color)
                 ),shape=16
             )+
  scale_x_log10()+
  scale_color_manual("Significance", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  # Tweaks
    scale_size_continuous(range = c(0.5,3))+

  ylab("Genetic diversity change (% π)")+
  scale_y_continuous(breaks = seq(-200,+400,by=50), labels = paste(seq(-200,+400,by=50),"%"))+
  coord_cartesian(ylim=c(-200,400))  +
  xlab(TeX("Time interval (y)"))+
  # xlab("")+
  theme_minimal(base_family = "Times New Roman")+
  # scale_x_reverse()+
  # theme(legend.position = "none", axis.text.x = element_blank())+
  theme(legend.position = "none")

plotc<-
ggplot(dfpici)+
  # geom_hline(yintercept = 0, color='darkgrey')+
  geom_segment(aes(y=dfpi$HedgesG - 1.96*dfpi$HedgesG.err,yend=dfpi$HedgesG +  1.96*dfpi$HedgesG.err,
                   # size=-log10(pvalue+1e-17),
                   x=dfpi$Gen.interval+0.1,
                   xend=dfpi$Gen.interval+0.1,
                   color=factor(pvalue_color)
                   ))+
  geom_point(aes(y=dfpi$HedgesG,
                 size=-log10(pvalue+1e-17),
                 x=dfpi$Gen.interval+0.1,
                 color=factor(pvalue_color)
                 ),shape=16
             )+
  scale_x_log10()+
  scale_color_manual("Significance", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  # Tweaks
    scale_size_continuous(range = c(0.5,3))+
  ylab("Hedges' g")+
  coord_cartesian(ylim=c(-5,5))  +
  xlab(TeX("Time Gen.interval (generations)"))+
  # xlab("")+
  theme_minimal(base_family = "Times New Roman")+
  # scale_x_reverse()+
  # theme(legend.position = "none", axis.text.x = element_blank())+
  theme(legend.position = "none")

plotd<-
ggplot(dfpici)+
  # geom_hline(yintercept = 0, color='darkgrey')+
  geom_segment(aes(y=Lower.CI,yend=Upper.CI,
                   # size=-log10(pvalue+1e-17),
                   x=dfpi$Gen.interval+0.1,
                   xend=dfpi$Gen.interval+0.1,
                   color=factor(pvalue_color)
                   ))+
  geom_point(aes(y=Percent.Change,
                 size=-log10(pvalue+1e-17),
                 x=dfpi$Gen.interval+0.1,
                 color=factor(pvalue_color)
                 ),shape=16
             )+
  scale_x_log10()+
  scale_color_manual("Significance", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  # Tweaks
  ylab("Genetic diversity change (% π)")+
  scale_size_continuous(range = c(0.5,3))+
  scale_y_continuous(breaks = seq(-200,+400,by=50), labels = paste(seq(-200,+400,by=50),"%"))+
  coord_cartesian(ylim=c(-200,400))  +
  xlab(TeX("Time Gen.interval (generations)"))+
  # xlab("")+
  theme_minimal(base_family = "Times New Roman")+
  # scale_x_reverse()+
  # theme(legend.position = "none", axis.text.x = element_blank())+
  theme(legend.position = "none")

plot_grid(plota,plotb, plotc,plotd)
```

```{r test_time_enrichment, echo=F , eval=F}
threshold=30

message("Negative trend enrichment")
table(
      as.numeric(dfpici$pvalue_color== (-1) ),
      dfpici$Gen.interval > threshold
      ) %>% fisher.test
message("Positive trend enrichment")
table(
      as.numeric(dfpici$pvalue_color== 1),
      dfpici$Gen.interval > threshold
      ) %>% fisher.test

message("Negative trend enrichment (against non-significant)")
table(as.numeric(dfpici[dfpici$pvalue_color!=1,]$pvalue_color == -1),
                 dfpici[dfpici$pvalue_color!=1,]$Gen.interval > threshold
                 ) %>% fisher.test()


  
```

```{r check_time_enrichment, echo=F , eval=T}
threshold=30


timeenrichmentexcludingpositive<-
  table(as.numeric(dfpici[dfpici$pvalue_color!=1,]$pvalue_color == -1),
                 dfpici[dfpici$pvalue_color!=1,]$Gen.interval > threshold
                 ) %>% fisher.test()

timetableall<-table(
      as.numeric(dfpici$pvalue_color != 0),
      dfpici$Gen.interval > threshold
      ) %>% fisher.test

timetablenegative<-table(
      as.numeric(dfpici$pvalue_color== (-1) ),
      dfpici$Gen.interval > threshold
      ) %>% fisher.test

timetablepos<-table(
      as.numeric(dfpici$pvalue_color== 1),
      dfpici$Gen.interval > threshold
      ) %>% fisher.test


#
# table(dfpici$Gen.interval>threshold, dfpici$pvalue_color)
# timeenrichment
# timeenrichmentpos
```

The number of significant decline cases in $\pi$ (`r sum(as.numeric(dfpici$pvalue_color == -1))`), were marginally significantly enriched with high number of generation interval (\>30 generations \# studies= `r sum(as.numeric(dfpici$Gen.interval > 30))`, Fisher's test $P$=`r timetablenegative$p.value`, Odds=`r timetablenegative$estimate`) (Fig. \ref{fig:timeenrich}). This was non-significant if excluding positive trends (which may not be realistic) ($P$=`r timeenrichmentexcludingpositive$p.value`, Odds `r timeenrichmentexcludingpositive$estimate`). Only focusing on studies with increasing genetic diversity trends showed also no enrichment (Fisher's test $P$=`r timetablepos$p.value`, Odds=`r timetablepos$estimate`).

```{r figexport_time, eval=F, warning=FALSE, echo=FALSE, include=FALSE}

tostore<-
ggplot(dfpici)+
  geom_segment(aes(y=Lower.CI,yend=Upper.CI,
                   # size=-log10(pvalue+1e-17),
                   x=dfpi$Gen.interval+0.1,
                   xend=dfpi$Gen.interval+0.1,
                   color=factor(pvalue_color)
                   ))+
  geom_point(aes(y=Percent.Change,
                 # size=-log10(pvalue+1e-17),
                 x=dfpi$Gen.interval+0.1,
                 color=factor(pvalue_color)
                 ),shape=16
             )+
  scale_x_log10()+
  scale_color_manual("Significance", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  # Tweaks
  ylab("Genetic diversity change (% π)")+
  scale_size_continuous(range = c(0.5,3))+
  scale_y_continuous(breaks = seq(-200,+400,by=50), labels = paste(seq(-200,+400,by=50),"%"))+
  coord_cartesian(ylim=c(-200,400))  +
  xlab(TeX("Time Gen.interval (generations)"))+
  # xlab("")+
  theme_minimal(base_family = "Times New Roman")+
  # scale_x_reverse()+
  # theme(legend.position = "none", axis.text.x = element_blank())+
  theme(legend.position = "none")+
  coord_cartesian(ylim=c(-100,100))+
    theme_minimal(base_family = "Times New Roman")+
    theme(
    panel.grid.major = element_blank(),  # remove major gridlines
    panel.grid.minor = element_blank(),   # remove minor gridlines
    # axis.line = element_blank()  # remove major gridlines
    axis.ticks = element_line(color = "grey"),     # turn ticks back on
    axis.ticks.length = unit(0.1, "cm")
  )+
  theme(legend.position = "none")


if(SAVEPLOTS){
  ggsave(filename = "figs/rawdata_vs_time.pdf",
            plot = tostore,
            height = 4,width = 4
            )
}

```

### Test differences with increasing data size.

```{r, warning=FALSE, error=F, include=T, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, fig.height=4, fig.width=3, fig.cap="Sample size (loci and individuals) and raw changes in π% genetic diversity."}

mytplot<-
dfpici %>%
  arrange(abs(pvalue_color)) %>%
ggplot()+
  geom_point(aes(y=Effective.N,x=N.Loci,
                size=abs(Percent.Change.truncated),
                color=factor(pvalue_color)
                 ),alpha=0.5
             )+
    scale_color_manual("Significance", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+

  scale_size_continuous("Abs. π%",range = c(0,4))+
  scale_x_log10()+
  scale_y_log10()+
  xlab("Number of genomic loci")+
  ylab("Number of individuals (Hn)")+
  theme_minimal(base_family = "Times New Roman")+
  theme(legend.position = "bottom")
mytplot

if(SAVEPLOTS){
  ggsave(filename = "figs/rawdata_vs_samplesizes.pdf",
            plot = mytplot,
            height = 4,width = 4
            )
}
```

```{r fig:sample_sizes ,echo=FALSE, warning=FALSE,message=FALSE,error=FALSE, eval=T, echo=FALSE, fig.height=3, fig.width=5, fig.cap="Sample numbers across past and present studies. \\label{fig:sample_sizes}"}
p1<-
ggplot(dfpici)+
   geom_jitter(aes(y=Early.Sample.N,
                x=factor(pvalue_color),
                color=factor(pvalue_color)
                 ),alpha=0.5
             )+
        stat_summary(aes(y=Early.Sample.N,
                x=factor(pvalue_color),
                color=factor(pvalue_color)
                 ),alpha=1, size=2
             )+
   geom_violin(aes(y=Early.Sample.N,
                x=factor(pvalue_color),
                color=factor(pvalue_color)
                 ),alpha=0.5
             )+
      scale_color_manual("Significance", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  scale_y_log10()+
  xlab("")+
  ylab("Number of individuals (early)")+
  theme_minimal(base_family = "Times New Roman")+
  theme(legend.position = "none")

p2<-
 ggplot(dfpici)+
    geom_jitter(aes(y=N.Loci,
                x=factor(pvalue_color),
                color=factor(pvalue_color)
                 ),alpha=0.5
             )+
      stat_summary(aes(y=N.Loci,
                x=factor(pvalue_color),
                color=factor(pvalue_color)
                 ),alpha=1, size=2
             )+
  geom_violin(aes(y=N.Loci,
                x=factor(pvalue_color),
                color=factor(pvalue_color)
                 ),alpha=0.5
             )+
      scale_color_manual("Significance", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  scale_y_log10()+
  ylab("Number of genomic loci")+
  xlab("")+
  theme_minimal(base_family = "Times New Roman")+
  theme(legend.position = "none")

 plot_grid(p1,p2,ncol=2)

```

```{r fig:sample_sizes_past_present ,echo=FALSE, warning=FALSE,message=FALSE,error=FALSE, eval=T, echo=FALSE, fig.height=3, fig.width=2.5, fig.cap="Sample and marker numbers across studies \\label{fig:samples_past_present}"}
tmp<-
dfpici %>% 
  pivot_longer(
    cols = c(Early.Sample.N, Recent.Sample.N),
    names_to = "Timepoint",
    values_to = "Sample.N"
  ) %>%
  mutate(Timepoint = recode(Timepoint,
                            "Early.Sample.N" = "Past",
                            "Recent.Sample.N" = "Present")) %>% 
  mutate(Sample.N= as.numeric(Sample.N))

ggplot(tmp)+
  geom_boxplot(aes(y=Sample.N,x=Timepoint))+
  # stat_summary(aes(y=Sample.N,x=Timepoint), size=2)+
  geom_jitter(aes(y=Sample.N,x=Timepoint,
                 ),alpha=0.5, color="grey"
             )+
  scale_y_log10()+
  ylab("Number of samples")+
  xlab("")+
  theme_minimal(base_family = "Times New Roman")+
  theme(legend.position = "none")

ttestsamplechange<- t.test(log10(as.numeric(tmp$Sample.N)) ~ factor(tmp$Timepoint))
wilcoxsamplechange<- wilcox.test(log10(as.numeric(tmp$Sample.N)) ~ factor(tmp$Timepoint))
meansamplechange=tapply( (as.numeric(tmp$Sample.N)) , factor(tmp$Timepoint), function(x) mean(x,na.rm=T))
meansamplechange=tapply( (as.numeric(tmp$Sample.N)) , factor(tmp$Timepoint), function(x) median(x,na.rm=T))
```

```{r check_samplesize_enrichment, echo=F , eval=T}

sizetable<-table(
      as.numeric(dfpici$N.Loci > 10),
      (dfpici$pvalue_color== -1)
      )
sizetableenrichment<-
  sizetable %>% fisher.test


sizetablepos<-table(
      as.numeric(dfpici$N.Loci > 10),
      (dfpici$pvalue_color== 1)
      )
sizetableenrichmentpos<-
  sizetablepos %>% fisher.test



ntable<-table(
      as.numeric(dfpici$Effective.N > 100),
      (dfpici$pvalue_color== -1)
      )
ntableenrichment<-
  ntable %>% fisher.test


ntablepos<-table(
      as.numeric(dfpici$Effective.N > 100),
      (dfpici$pvalue_color== 1)
      )
ntableenrichmentpos<-
  ntablepos %>% fisher.test

```

There was a significant enrichment of trajectories of $\pi$% decline in studies with number of loci above 10 (Odd=`r sizetableenrichment$estimate`, Fisher's test $P$=`r sizetableenrichment$p.value`) as well as $\pi$% increases (Odd=`r sizetableenrichmentpos$estimate`, Fisher's test $P$=`r sizetableenrichmentpos$p.value`).

There was not a significant enrichment of trajectories of $\pi$% decline in studies with \>100 individuals sampled (Odd=`r ntableenrichment$estimate`, Fisher's test $P$=`r ntableenrichment$p.value`) nor $\pi$% increases (Odd=`r ntableenrichmentpos$estimate`, Fisher's test $P$=`r ntableenrichmentpos$p.value`).

```{r check_data_size_averages, eval=T, warning=F, error=F,echo=FALSE,fig.width=3,fig.height=5, fig.cap="Average diversity change of different data sizes."}

averages <- dp %>% 
  mutate(samplesizegroup=2^round(log(N.Loci))) %>% 
  group_by(samplesizegroup) %>% 
  summarise(
    change = mean(Percent.Change, na.rm = TRUE),
    logchange = 100 * (exp(mean(Log.Change, na.rm = TRUE)) - 1),
    changeg = mean(Hedges.g.recalc, na.rm = TRUE),
    abundance = n(),
    changemeta=metaztest(Hedges.g.recalc, Hedges.g.SE.recalc)$Weighted_mean,
    changemetap=metaztest(Hedges.g.recalc, Hedges.g.SE.recalc)$p
  )
p1<-
ggplot(averages) +
  geom_point(aes(x = samplesizegroup, y = changemeta, color=factor(changemeta>0), size=abundance,shape=changemetap<0.05)) +
  ylim(-1.960,1.960)+
  scale_color_manual("", values = c(
    "FALSE"="red",
    "TRUE"="navy"
  ))+
    scale_shape_manual(values = c(1,16))+

  scale_x_log10()+
  ylab("Hedge's g average")+
  xlab("Number of loci (#)")+
  theme(legend.position = "none")


averages <- dp %>% 
  mutate(samplesizegroup=2^round(log(Early.Sample.N))) %>% 
  group_by(samplesizegroup) %>% 
  summarise(
    change = mean(Percent.Change, na.rm = TRUE),
    logchange = 100 * (exp(mean(Log.Change, na.rm = TRUE)) - 1),
    changeg = mean(Hedges.g.recalc, na.rm = TRUE),
    abundance = n(),
    changemeta=metaztest(Hedges.g.recalc, Hedges.g.SE.recalc)$Weighted_mean,
    changemetap=metaztest(Hedges.g.recalc, Hedges.g.SE.recalc)$p
  )

p2<-
 ggplot(averages) +
  geom_point(aes(x = samplesizegroup, y = changemeta, color=factor(changemeta>0), size=abundance, shape=changemetap<0.05)) +
  ylim(-1.960,1.960)+
  scale_color_manual("", values = c(
    "FALSE"="red",
    "TRUE"="navy"
  ))+
  scale_shape_manual(values = c(1,16))+
  scale_x_log10()+
  ylab("Hedge's g average")+
  xlab("Number of individuals (# early)")+
  theme(legend.position = "none")

plot_grid(p1,p2,ncol=1)
 
# averagescombined <- dp %>%
#    mutate(genomesizegroup=2^round(log(N.Loci))) %>%
#    mutate(samplesizegroup=2^round(log(Early.Sample.N))) %>%
#    group_by(samplesizegroup,genomesizegroup) %>%
#    summarise(
#      change = mean(Percent.Change, na.rm = TRUE),
#      logchange = 100 * (exp(mean(Log.Change, na.rm = TRUE)) - 1),
#      changeg = mean(Hedges.g.recalc, na.rm = TRUE),
#      abundance = n(),
#      changemeta=metaztest(Hedges.g.recalc, Hedges.g.SE.recalc)$Weighted_mean
#    )
# 
# p3<-
#  ggplot(averagescombined) +
#    geom_point(aes(y = samplesizegroup, x = genomesizegroup, color=factor(changemeta>0), size=abundance)) +
#    scale_x_log10()+
#    scale_y_log10()+
#    # ylim(-1.960,1.960)+
#    scale_color_manual("", values = c(
#      "FALSE"="red",
#      "TRUE"="navy"
#    ))+
#    scale_x_log10()+
#    ylab("Mean arithmetic genetic diversity change (%)")+
#    xlab("Number of individuals (#)")+
#    theme(legend.position = "none")
# 
#  
# plot_grid(plot_grid(p1,p2,ncol=1), p3)

```

### Test differences with decreasing noise.

```{r check_signi_variation, eval=T, warning=F, echo=FALSE,fig.width=6,fig.height=3, fig.cap="Relationship between raw values, significance and noise."}

# View 3
p1<-
ggplot(dfpici)+
  # geom_hline(yintercept = 0, color='darkgrey')+
  geom_segment(aes(y=Lower.CI,yend=Upper.CI,
                   # size=-log10(pvalue+1e-17),
                   x=SE.Percent.Change,
                   xend=SE.Percent.Change,
                   color=factor(pvalue_color)
                   ))+
  geom_point(aes(y=Percent.Change,
                 size=-log10(pvalue+1e-17),
                 x=SE.Percent.Change,
                 color=factor(pvalue_color)
                 ),shape=16
             )+
  scale_color_manual("Significance", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
    scale_size_continuous(range = c(0.5,3))+
  coord_cartesian(xlim=c(0,100),ylim=c(-100,100))+
  # add side histogram
  # geom_xsidehistogram(aes(x=-log10(pvalue+1e-17)), fill='darkgrey')+
  # Tweaks
  # scale_y_continuous(breaks = seq(-200,+400,by=50), labels = paste(seq(-200,+400,by=50),"%"))+
  # coord_cartesian(ylim=c(-200,400),)  +
  ylab("Percentage Change (π%)")+
  xlab(TeX("SE Percentage Change (%)"))+  # xlab("")+
  theme_minimal(base_family = "Times New Roman")+
  # scale_x_reverse()+
  # theme(legend.position = "none", axis.text.x = element_blank())+
  theme(legend.position = "none")



p2<-
  ggplot(dfpici)+
    # geom_hline(yintercept = 0, color='darkgrey')+
 geom_segment(aes(y=dfpi$HedgesG - 1.96*dfpi$HedgesG.err,yend=dfpi$HedgesG +  1.96*dfpi$HedgesG.err,
                     # size=-log10(pvalue+1e-17),
                     x=HedgesG.err,
                     xend=HedgesG.err,
                     color=factor(pvalue_color)
                     ))+
  geom_point(aes(y=dfpi$HedgesG,
                   size=-log10(pvalue+1e-17),
                   x=HedgesG.err,
                   color=factor(pvalue_color)
                   ),shape=16
               )+
    scale_color_manual("Significance", values = c(
                                                  "red",
                                                  "grey",
                                                  "navy"
                                                  ))+
    # add side histogram
    # geom_xsidehistogram(aes(x=-log10(pvalue+1e-17)), fill='darkgrey')+
    # Tweaks
    scale_size_continuous(range = c(0.5,3))+
    # coord_cartesian(xlim=c(0,100))+
    # scale_y_continuous(breaks = seq(-200,+400,by=50), labels = paste(seq(-200,+400,by=50),"%"))+
    # coord_cartesian(ylim=c(-200,400),)  +
    ylab("Hedges' g")+
    xlab(TeX("Hedges' g error"))+
    # xlab("")+
    theme_minimal(base_family = "Times New Roman")+
    # scale_x_reverse()+
    # theme(legend.position = "none", axis.text.x = element_blank())+
    theme(legend.position = "none")

plot_grid(p2,p1)
```

```{r fig:noise_updown ,echo=FALSE, warning=FALSE, eval=F}
ggplot(dfpici)+
  geom_boxplot(aes(y=SE.Percent.Change.delta,
                 x=factor(pvalue_color),
                 color=factor(pvalue_color))
  ,size=1)+
  geom_jitter(aes(y=SE.Percent.Change,
                 x=factor(pvalue_color),
                 color=factor(pvalue_color)
  ))+
  scale_y_log10()+
  scale_color_manual("Significance", values = c(
    "red",
    "grey",
    "navy"
  ))+
  xlab("")+
  theme_minimal(base_family = "Times New Roman")+
  theme(legend.position = "none")

# ggplot(dfpici)+
#   geom_boxplot(aes(y=HedgesG.err,
#                  x=factor(pvalue_color),
#                  color=factor(pvalue_color))
#   ,size=1)+
#   geom_jitter(aes(y=HedgesG.err,
#                  # size=-log10(pvalue+1e-17),
#                  x=factor(pvalue_color),
#                  color=factor(pvalue_color)
#   ))+
#   scale_y_log10()+
#   scale_color_manual("Significance", values = c(
#     "red",
#     "grey",
#     "navy"
#   ))+
#   xlab("")+
#   theme_minimal(base_family = "Times New Roman")+
#   theme(legend.position = "none")

```

```{r check_error_enrichm, echo=F , eval=T}
errortable<-table(
      as.numeric(dfpici$SE.Percent.Change < 5),
      (dfpici$pvalue_color== -1)
      )
errortableenrichment<-
  errortable %>% fisher.test
# errortableenrichment

errortablepos<-table(
      as.numeric(dfpici$SE.Percent.Change < 5),
      (dfpici$pvalue_color== 1)
      )
errortableenrichmentpos<-
  errortablepos %>% fisher.test
# errortableenrichmentpos

# # ta
# table(
#   as.numeric(dfpici$SE.Percent.Change < 5),
#   dfpici$Percent.Change>0
# ) %>% fisher.test()

```

It has become clear that the very extreme values of genetic diversity increases and decreases are owed to high standard error. A good rule of thumb is that standard error \< 5% are accurate measurements, while \>5% are highly uncertain, as the measurements we are aiming to achieve are in the same or smaller order of magnitude. We then look at enrichment of significant values in the more certain studies. I find a marginal enrichment of significant decline diversity cases $\pi$ and low error measurements (Odd=`r errortableenrichment$estimate`, Fisher's test $P$=`r errortableenrichment$p.value`) but not with significantly increasing $\pi$ studies (Odd=`r errortableenrichmentpos$estimate`, Fisher's test $P$=`r errortableenrichmentpos$p.value`).

### Test differences with domesticated vs not species.

```{r, eval=T, echo=FALSE}
# Relationship domesticated and negative trend
# table(dp$pvalue_color== -1,
#       !is.na(d$Domestic.Pest.Pathogen.Status)
#       ) %>% fisher.test()

# Relationship domesticated and positive trend
fisherdomesticated<-
table(dp$pvalue_color== 1,
      !is.na(d$Domestic.Pest.Pathogen.Status)
      ) %>% fisher.test()

# mean(dp$Percent.Change[!is.na(d$Domestic.Pest.Pathogen.Status)], na.rm=T)
# mean(dp$Percent.Change[is.na(d$Domestic.Pest.Pathogen.Status)], na.rm=T)
```

Finally, hat the domesticated/pathogens/pest speices not being removed, we would find an upward bias in genetic diversity change as there is an enrichment of domesticated species and significant positive trends in the original data (Odds= `r fisherdomesticated$estimate`, Odds= `r fisherdomesticated$p.value`).

\clearpage

## Text S3: Test within-species consistency.

There were repeated measures per species in the dataset. For instance, for $\pi$ there were `r nrow(dfpi)` studies, but withihn these there were a total of `r length(unique(dfpi$Binomen.rotl))` unique species.

We hypothesize that if the observed decline are robus they should occur in the same direction within the same species We then conduct a Binomial test of species with multiple significant observations, asking whether they tend to have more positive or more negative significant values.

```{r fig:check_signi_populations, eval=T, warning=FALSE, echo=FALSE,fig.width=3,fig.height=3, fig.cap="Cases of species with multiple observations. \\label{fig:multipleobservations}"}
dfpi<-
  dfpi %>%
  dplyr::group_by(Binomen.rotl) %>%
  dplyr::mutate(Num_obs_per_species=length(Binomen.rotl)) #%>%
  # dplyr::select(Num_obs_per_species) %>% arrange %>%  head # debug

ggplot(dfpici)+
  geom_segment(aes(y=Lower.CI,yend=Upper.CI,
                 x=dfpi$Num_obs_per_species,
                 xend=dfpi$Num_obs_per_species,
                   color=factor(pvalue_color)
                   ))+
  geom_point(aes(y=Percent.Change,
                 x=dfpi$Num_obs_per_species,
                 color=factor(pvalue_color)
                 ),shape=16
             )+
  scale_x_log10()+
  scale_color_manual("Significance", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  # Tweaks
  ylab("Genetic diversity change (% π)")+
  scale_y_continuous(breaks = seq(-200,+400,by=50), labels = paste(seq(-200,+400,by=50),"%"))+
  coord_cartesian(ylim=c(-200,400))  +
  xlab(TeX("Number of observations per species"))+
  # xlab("")+
  theme_minimal(base_family = "Times New Roman")+
  # scale_x_reverse()+
  # theme(legend.position = "none", axis.text.x = element_blank())+
  theme(legend.position = "none")


```

```{r check_signi_populations_sides, eval=T, warning=FALSE, echo=FALSE}
# Get th number of significant tests per species

dfpici$Num_obs_per_species<-dfpi$Num_obs_per_species
dfpici$Binomen.rotl<-dfpi$Binomen.rotl

dfpi_perspecies<-
dfpici %>%
  dplyr::group_by(Binomen.rotl) %>%
  dplyr::summarise(
                Num_obs_per_species=head(Num_obs_per_species,1),
                num_signi= sum(abs(pvalue_color)),
                num_positive= sum((pvalue_color[pvalue_color>0])),
                num_negative= sum((pvalue_color[pvalue_color<0]))
  )

# Where the multiple observations of the same paper?
repeatedmeasures<-
dfpici %>%
  # dplyr::filter(pvalue_color != 0) %>%
  group_by(Binomen.rotl) %>%
  summarise(independentobservations=length(unique(PaperID)))

# table(repeatedmeasures$independentobservations>1)
```

```{r check_signi_repeatability, evaluate=T, warning=FALSE,echo=F, message=F, error=FALSE, fig.width=3.5,fig.height=3, fig.cap="Positive and negative cases of species with multiple observations. \\label{fig:multipleobservationspositivenegative}"}

tmp<-
  dfpi_perspecies %>%
  dplyr::filter(num_signi>=1)

Binom_est<-apply(tmp,1,FUN = function(i){
  suc=as.numeric(i[4]) %>% abs
  tot=as.numeric(i[3]) %>% abs
  binom.test(x=suc, n=tot, p = 0.5, alternative = "less")$estimate
  })

Binom_p<-apply(tmp,1,FUN = function(i){
  suc=as.numeric(i[4])
  tot=as.numeric(i[3])
  binom.test(x=suc, n=tot, p = 0.5, alternative = "less")$p.value
  })

tmp$Binom_est<-Binom_est
tmp$Binom_p<-Binom_p

tmp<-
  tmp %>% arrange(Binom_p)
ggplot(tmp)+
  geom_jitter(aes(y=num_positive,x=num_negative,color=Binom_p<0.05),width = 0.1, height = 0.1,shape=16)+
  scale_x_reverse()+
  scale_color_manual("50/50 rejected",values = c("TRUE"="red","FALSE"="grey"))+
  ylab(TeX("Number of significant positive trends"))+
  xlab(TeX("Number of significant negative trends"))+
  coord_equal()+
  theme_minimal(base_family = "Times New Roman")#+
  # theme(legend.position = "none")

```

Of `r nrow(tmp)` species with multiple populations sampled and significant trends, none had a ratio of negative trends (decline) vs positive trend (increase) that was significant under a Binomial test of expected 50/50% probability.

\clearpage

## Text S4: Interpretation of changes under an evolutionary population genetic model of species contraction.

Below we explore what % genetic diversity changes (non-significant or significant) could mean in terms of population contraction models of different scenarios (we assume positive changes are zero, as they are treated separately, see Text S5). Equations are derived in the **Mathematical Appendix**.

Some key notation:\
- $\pi$ nucleotide diversity or expected heterozygosity (which are identical under a biallelic loci condition).\
- $M$ allelic richness, or segregating sites, or number of variable positions, or number of mutations in a strand of DNA. Note we use $M$ instead of more standard $S$ or $A$ to avoid confusions with species richness from ecology, and also to avoid confusions with area of habitat.\
- $N_0$ the population size at some past time. We may also use $N$ as this is the long-term equilibrium size.\
- $N_t$ the population size at a given time $t$. Because we mostly deal with two time points (0 and 1), we often use $N_1$ the population size instantaneous reduction from human impact or contraction.\
- $A_0$ the habitat area of a species with multiple populations (useful to use area to avoid defining discrete entities).\
- $A_1$ the habitat area of a species reduced from human impact or contraction.\
- $X_{\pi}$ the fraction loss of genetic diversity $\pi$. Intuitive % of genetic diversity loss when $X_{\pi} \times 100$.\
- Same for $X_M$ the fraction loss of allelic richness $M$.\

### BIOLOGICAL SAMPLING -- *null scenario*

#### *Average genetic diversity* $\pi$

```{r load simulations, echo=FALSE,error=F,eval=T, message=F}
RELOAD=F

if(RELOAD){
  # Load ms simulation
  # sims<-read.csv("Response-GenDivMetaAnalysis/pi_grid_simulation_results.csv")
  # sims<-read.csv("pi_grid_simulation_results_realistic.csv")
  sims<-read.csv("pi_grid_simulation_results_realistic_loci.csv")
  # Modify and merge with subset
  dfpici<-
    dfpici %>% 
    dplyr::mutate(n1=Recent.Sample.N,
                   n0=Early.Sample.N
                   )
  # Merge with subset
  m<-
    merge(
      dfpici,sims,
      all.x=T
    ) %>% 
    dplyr::filter(totalbp==num_sites)
    ms<-m
    save(file = "tmp/ms.rda", ms)
}else{
  load(file = "tmp/ms.rda")
}
```

```{r null plot, evaluate=F, warning=FALSE,echo=F, message=F, error=FALSE, fig.width=8,fig.height=8, fig.cap="Ranges of expected changes of diversity π% with biological sampling \\label{fig:biological_sampling_ranges}"}

allnullplot<-
  ms %>% 
  group_by(n0,n1, N.Loci,Percent.Change,Lower.CI,Upper.CI,pvalue_color,Binomen.rotl, observation, PaperID) %>% 
  summarise(relative_change_min=quantile(relative_change, 0.05, na.rm = T),
            relative_change_max=quantile(relative_change, 0.95, na.rm = T),
            relative_change_mean=quantile(relative_change, 0.5, na.rm = T),
            n=length(relative_change),
            
  ) %>% 
  arrange(abs(pvalue_color)) %>% 
  ggplot(.)+
  geom_segment(aes(
    y=relative_change_min, yend=relative_change_max,
    x=Percent.Change, xend=Percent.Change,
    color=factor(pvalue_color)
  ))+
  geom_segment(aes(
    y=relative_change_mean, yend=relative_change_mean,
    x=Lower.CI, xend=Upper.CI,
    color=factor(pvalue_color)
  ))+
  geom_point(aes(y=relative_change_mean, x=Percent.Change,
                 color=factor(pvalue_color)))+
  ggrepel::geom_text_repel(
    aes(y = relative_change_mean, x = Percent.Change, label = Binomen.rotl, color = factor(pvalue_color)),
    size = 3,
    family = "Times New Roman",
    fontface = "italic",
    max.overlaps = 100,
    force = 5,           # increase repelling strength
    force_pull = 0.1,      # keep connection to origin
    nudge_x = 1         # nudge outward (you can try nudge_y = 2 instead or as well)
  )+
  scale_color_manual("Significance", values = c("red","grey","navy"))+
  geom_abline(alpha=0.5)+
  coord_cartesian(xlim=c(-50,50), ylim=c(-50,50))+
  xlab("Percent change (%π)")+ylab("Null expectation of change (%π)")


allnullplot

if(SAVEPLOTS){
  ggsave(filename = "figs/null_scenario-rangesplot.pdf",
         plot = allnullplot,
         height = 12,width = 12
  )
}
```

The high noise is likely due to the small sampling sizes:\
Present sampling median n0=`r printmedian(as.numeric(dfpicisubset$Early.Sample.N))`\
Present sampling median n1=`r printmedian(as.numeric(dfpicisubset$Recent.Sample.N))`\
Number of markers median L=`r printmedian(as.numeric(dfpicisubset$totalbp))`\

```{r fig:biological_sampling ,echo=FALSE, warning=FALSE,message=FALSE,error=FALSE, eval=T, echo=FALSE,fig.width=5,fig.height=8, fig.cap="Expected maximum changes of diversity π% with biological sampling \\label{fig:biological_sampling}"}

# Summarize max values
mss<-
  ms %>% 
  group_by(n0,n1, N.Loci,Percent.Change,pvalue_color,Binomen.rotl, observation) %>% 
  summarise(maxchange=quantile(relative_change, 0.05, na.rm = T),n=length(relative_change))
maxss<-
  ms %>% 
  group_by(n0,n1, N.Loci,Percent.Change,pvalue_color,Binomen.rotl, observation) %>% 
  summarise(maxchange=quantile(relative_change, 0.95, na.rm = T),n=length(relative_change))
  

plota<-
ggplot(mss)+
  geom_point(aes(y=maxchange, x=Percent.Change, color=factor(pvalue_color)), bins=50)+
  xlim(c(-50,0)) + ylim(c(-50,0))+
  geom_abline()+
  scale_color_manual("Significance", values = c(
    "red",
    "grey",
    "navy"
  ))+
  ggrepel::geom_text_repel(
    # data=dplyr::filter(mss, Percent.Change< - 5), 
    # data=dplyr::filter(mss, pvalue_color !=0),
    data=dplyr::filter(mss, Percent.Change < -5, maxchange < -5),
    aes(y=maxchange, x=Percent.Change,label=Binomen.rotl, color=factor(pvalue_color)),
    size = 3,                            # Smaller font size
    family = "TimesNewRoman",          # Use Times New Roman
    fontface = "italic" ,         # Use Times New Roman
    max.overlaps = 25,
    force = 5,           # increase repelling strength
    force_pull = 0.1,      # keep connection to origin
    nudge_x = 1         # nudge outward (you can try nudge_y = 2 instead or as well)
    ) +
  theme_minimal()+
  xlab("Percent change (%π)")+ylab("Null expectation of change (%π)")+
  theme_minimal(base_family = "Times New Roman")
# plota

plotb<-
  ggplot(maxss)+
  geom_point(aes(y=maxchange, x=Percent.Change, color=factor(pvalue_color)), bins=50)+
  xlim(c(0,50)) + ylim(c(0,50))+
  geom_abline()+
  scale_color_manual("Significance", values = c(
    "red",
    "grey",
    "navy"
  ))+
  ggrepel::geom_text_repel(
    # data=dplyr::filter(maxss, Percent.Change> 5),
    # data=maxss,
    # data=dplyr::filter(maxss, pvalue_color !=0),
    data=dplyr::filter(maxss, Percent.Change> 5, maxchange>5),
    aes(y=maxchange, x=Percent.Change,label=Binomen.rotl, color=factor(pvalue_color)),
       size = 3,                            # Smaller font size
    family = "TimesNewRoman",          # Use Times New Roman
    fontface = "italic" ,         # Use Times New Roman
    max.overlaps = 25,
    force = 5,           # increase repelling strength
    force_pull = 0.1,      # keep connection to origin
    nudge_x = 1         # nudge outward (you can try nudge_y = 2 instead or as well)
  ) +
  theme_minimal()+
  xlab("Percent change (%π)")+ylab("Null expectation of change (%π)")+
  theme_minimal(base_family = "Times New Roman")


biosamplingplot<-
  plot_grid(plotb+theme(legend.position = "none"), 
            plota+theme(legend.position = "none"), ncol=1)
biosamplingplot

if(SAVEPLOTS){
# if(T){
  ggsave(filename = "figs/null_scenario.pdf",
            plot = biosamplingplot,
            height = 8,width = 4
            )
}
```

```{r possible new plot ,echo=FALSE, warning=FALSE,message=F, error=FALSE, eval=T, echo=FALSE}

ggsave(filename = "figs/null_scenario.pdf",
       plot = plot_grid(plotb+theme(legend.position = "none")+coord_flip(), 
                        plota+theme(legend.position = "none")+coord_flip(), 
                        ncol=1),
       height = 8,width = 4
)

```

```{r biological_sampling ,echo=FALSE, warning=FALSE,message=F, error=FALSE, eval=T, echo=FALSE}

negativesamplings<-
  table(mss$Percent.Change>mss$maxchange)


dfpicisubset_empiricalp<-
  ms %>% 
  filter(pvalue_color== -1) %>%
  group_by(n0,n1, N.Loci,Percent.Change,pvalue_color,Binomen.rotl,PaperID) %>% 
  summarise(pempirical=sum(Percent.Change<relative_change)/length(relative_change),
            n=length(relative_change))

# table(dfpicisubset_empiricalp$pempirical<0.05)/nrow(dfpicisubset_empiricalp)
# 
meanbeyondsampling<-
  dfpicisubset_empiricalp$Percent.Change[dfpicisubset_empiricalp$pempirical<0.05]

extremestudies<-
  mss$Percent.Change[mss$Percent.Change<mss$maxchange] %>% length
extrememeanchange<-
  mss$Percent.Change[mss$Percent.Change<mss$maxchange] %>% printmean()
extrememeanchangenull<-
  mss$maxchange[mss$Percent.Change<mss$maxchange] %>% printmean()
extrememeanchangeexcess<-
  (mss$Percent.Change-mss$maxchange)[mss$Percent.Change<mss$maxchange] %>% printmean()


```

Taking biological sampling into account, we found `r extremestudies` studies out of `r nrow(mss)` with changes in measured genetic diversity (`r extrememeanchange`%) beyond the 5% tail of the null distribution from subsampling (`r extrememeanchangenull`%). This corresponds to an excess of `r extrememeanchangeexcess`%.

```{r null_expectation_sampling, echo=F, warning=F,error=F, eval=T, message=F}
REDO=F

if(REDO){
  changesamples<-c()
  samples=1000
  
  for( i in 1:samples){
    tmp<-
      ms %>% 
      dplyr::filter(!is.na(relative_change)) %>% 
      dplyr::group_by(observation) %>% 
      dplyr::sample_n(size = 1) %>% 
      ungroup %>% 
      dplyr::summarise(mean=mean(relative_change,na.rm=T)) %>% unlist
    
    changesamples[i]<-tmp
  }
  
  save(file = "tmp/changesamples.rda", changesamples)
}else{
  load("tmp/changesamples.rda")
}

```

Bootstrapping 1000 times one simulation per paper gives a range of average π% change= `r printranges(changesamples)`%.

The reason observations are very noisy is because the median sampling effort in Shaw is $n_0$=`r printmedian(as.numeric(dfpi$Early.Sample.N))` ,$n_1$=`r printmedian(as.numeric(dfpi$Recent.Sample.N))`, and the median number of loci is $L$=`r printmedian(as.numeric(dfpi$N.Loci))`. Fig. \ref{fig:SAMPLE_panmictic_noise_grid} shows the results of standard deviation of percent diversity change (%π) in a population of constant size sampled two times, in a grid of values from 5 individuals to 10,000 individuals, and a number of markers 1 to 1M. The median values of Shaw are marked in green, and indicate that the median error rate is over 10%, just as we see in the data.

```{r minimum_sample_size,  echo=F, warning=F,error=F, eval=T, message=F, fig.width=5,fig.height=3.5, fig.cap="Expected noise in percent diversity change π% with different ssample sizes \\label{fig:null_sampling}"}
# sims<-read.csv("pi_grid_simulation_results_realistic_loci.csv")
sims<-read.csv("pi_grid_simulation_results.csv")

testsims<-
  sims %>% 
  # mutate(relative_change=relative_change-mean(relative_change,na.rm=T)) %>% 
  dplyr::filter(n0 >2, L>2) %>% 
  group_by(L,n0,n1) %>% 
  summarise(
    meanchange=mean((relative_change), na.rm=T),
    abschange=mean(abs(relative_change), na.rm=T),
    sdschange=sd(abs(relative_change), na.rm=T)
    )


p1<-
ggplot(testsims)+
  geom_point(aes(y=sdschange, x=n0), color="grey")+
  stat_summary(aes(y=sdschange, x=n0))+
  xlim(c(1,1e6+1))+
  ylim(c(-50,50))+
  scale_y_log10()+
  scale_x_log10(breaks=10^(0:6))+
  ylab("SD percent diversity change (%π)")+
  xlab(TeX("Sample size ($n_0$ = $n_1$)"))+
  theme_minimal(base_family = "Times New Roman") +
  geom_vline(xintercept = 40, color="#6ABD45")


p2<-
ggplot(testsims)+
  geom_point(aes(y=sdschange, x=L), color="grey")+
  stat_summary(aes(y=sdschange, x=L))+
  xlim(c(1,1e6+1))+
  ylim(c(-50,50))+
  scale_y_log10()+
  scale_x_log10(breaks=10^(0:6))+
  ylab("SD percent diversity change (%π)")+
  xlab(TeX("Numer of markers ($L$)"))+
  theme_minimal(base_family = "Times New Roman") +
  geom_vline(xintercept = 10, color="#6ABD45")

plot_grid(p1,p2)

valueerrormedian<-
testsims %>% 
  dplyr::filter(n0 <= quantile(as.numeric(dfpici$Early.Sample.N), 0.5, na.rm=T)) %>% 
  dplyr::filter(L <= quantile(as.numeric(dfpici$N.Loci), 0.5, na.rm=T)) %>% 
  dplyr::filter(!is.infinite(sdschange)) %>% 
  pull(sdschange) %>% 
  min(na.rm=T)
valueerror75<-
testsims %>% 
  dplyr::filter(n0 <= quantile(as.numeric(dfpici$Early.Sample.N), 0.75, na.rm=T)) %>% 
  dplyr::filter(L <= quantile(as.numeric(dfpici$N.Loci), 0.75, na.rm=T)) %>% 
  dplyr::filter(!is.infinite(sdschange)) %>% 
  pull(sdschange) %>% 
  min(na.rm=T)

neededsample1perc<-
  testsims %>% 
  dplyr::filter(sdschange <=1) %>% 
  pull(n0) %>% 
  min(na.rm = T)
neededseq1perc<-
  testsims %>% 
  dplyr::filter(sdschange <=1) %>% 
  pull(L) %>% 
  median(na.rm = T)

  
  
```

Based on Fig. S\ref{fig:null_sampling}, the expected error for 50% of the Shaw dataset is a minimum of `r valueerrormedian`%, and for 75% percentile of the dataset a minimum of `r valueerror75`%.

#### *Allelic richness* $M$  

TBD.

\clearpage

### SUBSAMPLE Single population, immediate -- *scenario 1a*

#### *Average genetic diversity* $\pi$  

Based on population evolutionary genetic theory, in the **Mathematical Appendix**, we show that the expected relative $\pi$% change (or the fraction change, $X_\pi$, which is more mathematically convenient) from a reduction of $N_0$ to $N_1$ within a generation is null. The only reason genetic diversity may change as a subsample of a population is due to sampling noise or if for some reason the remaining fraction of the population is more related to each other. This affects a fraction of the dataset: `r table(dfpici$Gen.interval<1)['TRUE']` observations, `r clean(100*table(dfpici$Gen.interval<1)["TRUE"]/nrow(dfpici))`%. The results in Fig. \ref{fig:SAMPLE_panmictic_noise_grid}, corroborate the classic observation that substantial losses (\~5%) of genetic diversity are not expected unless the bottleneck in a species is \<\<100 individuals. NB. Fig. \ref{fig:SAMPLE_panmictic_noise_grid}: A jitter of 0.1N units was added for visualization. Min value of bottleneck truncated to 10M. Positive trend % of genetic diversity collapsed to 0%.\

```{r scenario1a_test ,echo=FALSE, warning=FALSE, eval=F}
res=G_inv_drift_population_loss(dfpicisubset,
                                # generations = round(dfpicisubset$Gen.interval+1)
                                generations = rep(1,nrow(dfpicisubset))
                                )
res$Gen.interval<-round(dfpicisubset$Gen.interval+1)
res$Gen.interval[res$Gen.interval>100]<-100
res$pvalue_color<-dfpicisubset$pvalue_color
res$Binomen.rotl<-dfpicisubset$Binomen.rotl
res<-res %>% arrange(abs(pvalue_color))

printmedian(res$Pop.Bottleneck[res$pvalue_color== -1 &
                                 res$Gen.interval<=1
                                 ])
printIQR(res$Pop.Bottleneck[res$pvalue_color== -1 &
                                 res$Gen.interval<=1
                                 ])

```

```{r SAMPLE_panmictic, eval=T, echo=FALSE,warning=FALSE, fig.width=4,fig.height=3.5}

# infer population bottleneck based on panmictic expectation
res=G_inv_immediate_population_loss_panmictic(dfpicisubset)
# Repeat with the 1/2N approach
res=G_inv_drift_population_loss(dfpicisubset,
                                generations = rep(1,nrow(dfpicisubset))
                                )
# add the p values from Hedge's tets
res$pvalue_color<-dfpicisubset$pvalue_color
res$Binomen.rotl<-dfpicisubset$Binomen.rotl
res<-res %>% arrange(abs(pvalue_color))

# plot
myplot <-
  ggplot(res) +
  geom_jitter(aes(y = Percent.Change, x = -log10(Pop.Bottleneck), color = factor(pvalue_color)),
              height = 1, width = 0.1, alpha = 0.5, shape = 16) +
  scale_color_manual("Trend", values = c("red", "grey", "navy")) +
  xlab("Population immediate bottleneck (N)") +
  ylab("Percent diversity change (π%)") +
  scale_x_continuous(
    breaks = -6:-1,
    limits = c(-6, -1),
    labels = TeX(sprintf("$10^{%d}$", abs(-6:-1)))
  ) +
  ggrepel::geom_text_repel(
    data = dplyr::filter(res, Pop.Bottleneck < 100),
    aes(y = Percent.Change, x = -log10(Pop.Bottleneck), color = factor(pvalue_color), label = Binomen.rotl),
    size = 3,                            # Smaller font size
    family = "TimesNewRoman" ,         # Use Times New Roman
    fontface = "italic"
  ) +
  theme_minimal(base_family = "Times New Roman") +
  ylim(-100, 0)

# myplot


if(SAVEPLOTS){
  ggsave(filename = "figs/scenario1a_pi_nonoise.pdf",
            plot = myplot,
            height = 4,width = 4
            )
}


replicates=100

# Wrapper for conciseness
datsim<-wrapper_model(dfpici,replicates,"Pop.Bottleneck", G_inv_immediate_population_loss_panmictic) %>%
  arrange(abs(pvalue_color))

# plot
myplot2<-
ggplot(datsim)+
  geom_point(aes(y=Percent.Change.original,x=-log10(Pop.Bottleneck), color=factor(pvalue_color)),
              height = 1, width = 0.1, alpha=0.5, shape=16)+
    scale_color_manual("Trend", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  xlab("Population immediate bottleneck (N)")+
  ylab("Percent diversity change (π%)")+
  scale_x_continuous(
    breaks = -4:-1,
    limits = c(-4, -1),
    labels = TeX(sprintf("$10^{%d}$", abs(-4:-1)))
  )+
  ylim(-100,0)+
  theme_minimal(base_family = "Times New Roman")

# myplot2

if(SAVEPLOTS){
  ggsave(filename = "figs/scenario1a_pi_noise.pdf",
            plot = myplot2,
            height = 4,width = 4
            )
}
```

```{r fig:SAMPLE_panmictic_noise_grid ,echo=FALSE, warning=FALSE, eval=T, warning=FALSE, echo=FALSE,fig.width=7,fig.height=3.5, fig.cap="Expected immediate population bottleneck given observed π% with noise \\label{fig:SAMPLE_panmictic_noise_grid}"}

plot_grid(myplot,myplot2)

```

*No noise*  <!-- Results (without error sampling): n = `r length((res$Pop.Bottleneck[res$pvalue_color== -1]))` observartions.\ --> Median = `r printmedian(res$Pop.Bottleneck[res$pvalue_color== -1])` individuals.\
Mean = `r printmean(res$Pop.Bottleneck[res$pvalue_color== -1])` individuals.\
IQR = `r printIQR(res$Pop.Bottleneck[res$pvalue_color== -1])` individuals.\
Range = `r printranges(res$Pop.Bottleneck[res$pvalue_color== -1])` individuals.\

*Noise*  <!-- n = `r length((datsim$Pop.Bottleneck[datsim$pvalue_color== -1]))` observations modeled.\ --> Median = `r printmedian(datsim$Pop.Bottleneck[datsim$pvalue_color== -1 & datsim$Pop.Bottleneck!=10000])` individuals.\
Mean = `r printmean(datsim$Pop.Bottleneck[datsim$pvalue_color== -1 & datsim$Pop.Bottleneck!=10000])` individuals.\
IQR = `r printIQR(datsim$Pop.Bottleneck[datsim$pvalue_color== -1 & datsim$Pop.Bottleneck!=10000])` individuals.\
Range = `r printranges(datsim$Pop.Bottleneck[datsim$pvalue_color== -1 & datsim$Pop.Bottleneck!=10000])` individuals.\

#### *Allelic richnes* $M$  

```{r fig:MAR_panmictic_noise ,echo=FALSE, warning=FALSE, eval=T,fig.width=7,fig.height=3.5, fig.cap="Expected immediate population bottleneck given observed M%. \\label{fig:SAMPLE_panmictic_noise_grid_M}"}
SAVEPLOTS=F

# infer population bottleneck based on panmictic expectation
res1m=M_inv_immediate_population_loss_panmictic(dfmci,1e6) %>%
      mutate(N0=1e6)
res10k=M_inv_immediate_population_loss_panmictic(dfmci,10000)%>%
      mutate(N0=10000)
res1k=M_inv_immediate_population_loss_panmictic(dfmci,1000)%>%
      mutate(N0=1000)
res100=M_inv_immediate_population_loss_panmictic(dfmci,100)%>%
      mutate(N0=100)

# bind the inferences
res<-
  rbind(res1m,res10k,res1k,res100)

# add the p values from Hedge's tets
res$pvalue_color<-rep(dfmci$pvalue_color,4)
res<-res %>% arrange(abs(pvalue_color))

# plot
myplot<-
ggplot(res)+
  geom_jitter(aes(y=Percent.Change,x=Pop.Bottleneck, shape=factor(N0),color=factor(pvalue_color)),
              height = 1, width = 1, alpha=0.5)+
    scale_color_manual("Trend", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  scale_shape(TeX("$N_0$"))+
  xlab("Population bottleneck (%)")+
  ylab("Percent diversity change (M%)")+
  xlim(0,100)+ylim(-100,0)+
  theme_minimal(base_family = "Times New Roman")
# myplot

if(SAVEPLOTS){
  ggsave(filename = "figs/scenario1a_m_nonoise.pdf",
            plot = myplot,
            height = 4,width = 4
            )
}

# Set assumed starting population
N0=10000
replicates=100

# Wrapper for conciseness
datsim<-wrapper_model(dfpici,replicates,"Pop.Bottleneck", M_inv_immediate_population_loss_panmictic , N0) %>%
  arrange(abs(pvalue_color))

# plot
myplot2<-
ggplot(datsim)+
  geom_jitter(aes(y=Percent.Change.original,x=Pop.Bottleneck, color=factor(pvalue_color), shape=N0),
              height = 1, width = 1, alpha=0.5,shape=16)+
    scale_color_manual("Trend", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  xlab("Population bottleneck (%)")+
  ylab("Percent diversity change (M%)")+
  xlim(0,100)+ylim(-100,0)+
  theme_minimal(base_family = "Times New Roman")
# myplot

if(SAVEPLOTS){
  ggsave(filename = "figs/scenario1a_m_nonoise.pdf",
            plot = myplot,
            height = 4,width = 4
            )
}


# To report example of 10K start
res<-res10k

# Plot final grid
plot_grid(myplot,myplot2)

```

The loss of alleles $M$ follows under the coalescent a function of: $X_{M} = -\frac{\log(1 - X_{N})}{\log(N_0)}$. In the case of alleles $M$ the loss not only depends on the fraction loss of individuals $X_N$, but also the starting population sizes $N_0$ (this is unknown, so Fig. \ref{fig:Immediate_panmictic_noise_grid} shows trajectories with several $N_0$). NB Fig. \ref{fig:Immediate_panmictic_noise_grid} A jitter of 1N units was added for visualization. Positive trend % of genetic diversity collapsed to 0%. The patter observed of four trajectories correspond to inferences assuming $N_0$= 100, 1000, 10000, 1000000 starting individuals prior the bottleneck (respectively left to right).\

Using the results assuming a starting size of $N_0$=10000 individuals. For allelic richness $M$ the loss is faster than for $\pi$ diversity, as expected.

*No noise* (assume the case of $N_0$=10000)  Median = `r printmedian((res$Pop.Bottleneck[res$pvalue_color==-1]))`% of original population\
Mean = `r printmean((res$Pop.Bottleneck[res$pvalue_color==-1]))`% of original population\
IQR = `r printIQR((res$Pop.Bottleneck[res$pvalue_color==-1]))`% of original population\
IQR = `r printranges((res$Pop.Bottleneck[res$pvalue_color==-1]))`% of original population\

*Noise* (assume the case of $N_0$=10000)  Median = `r printmedian((datsim$Pop.Bottleneck[datsim$pvalue_color==-1]))`% of original population\
Mean = `r printmean((datsim$Pop.Bottleneck[datsim$pvalue_color==-1]))`% of original population\
IQR = `r printIQR((datsim$Pop.Bottleneck[datsim$pvalue_color==-1]))`% of original population\
IQR = `r printranges((datsim$Pop.Bottleneck[datsim$pvalue_color==-1]))`% of original population\

\clearpage

### TIME Single population, drift over time -- *scenario 1b*

#### *Average genetic diversity* $\pi$

```{r fig:drift_pi_noise ,echo=FALSE, warning=FALSE, eval=T,fig.width=7,fig.height=3.5, fig.cap="Expected population bottleneck and drift given observed π%. \\label{fig:Drift_panmictic_noise_grid}"}

res=G_inv_drift_population_loss(dfpicisubset,generations = round(dfpicisubset$Gen.interval+1))
res$Gen.interval<-round(dfpicisubset$Gen.interval+1)
res$Gen.interval[res$Gen.interval>100]<-100
res$pvalue_color<-dfpicisubset$pvalue_color
res$Binomen.rotl<-dfpicisubset$Binomen.rotl
res<-res %>% arrange(abs(pvalue_color))

res$Pop.Bottleneck[is.infinite(res$Pop.Bottleneck)]<- 1e6
res$Pop.Bottleneck[res$Pop.Bottleneck>1e6]<-1e6

# plot
myplot<-
ggplot(res)+
  geom_point(aes(y=Percent.Change,
                 x=-log10(Pop.Bottleneck),
                 size=Gen.interval,
                 color=factor(pvalue_color)  ),
             # height = 1, width = 1,
             alpha=0.5,shape=16,shape=16)+
  scale_color_manual("Trend", values = c(
    "red",
    "grey",
    "navy"
  ))+
  # ggrepel::geom_text_repel(
  #   data = res,
  #   aes(y=Percent.Change,
  #       x=-log10(Pop.Bottleneck),
  #       color=factor(pvalue_color), 
  #       label = Binomen.rotl),
  #  family = "Times New Roman",
  #     fontface = "italic",
  #     max.overlaps = 100,
  #     force = 5,           # increase repelling strength
  #     force_pull = 0.1,      # keep connection to origin
  #     nudge_x = 1         # nudge outward (you can try nudge_y = 2 instead or as well)
  # ) +
  theme_minimal(base_family = "Times New Roman") +
  xlab(TeX("Population with drift (N)"))+
  ylab("Percent diversity change (π%)")+
  scale_size_continuous("Generations",range = c(0.5,4), breaks = c(1,25,50,75))+
  scale_x_continuous(
    breaks = -6:-1,
    limits = c(-6, -1),
    labels = TeX(sprintf("$10^{%d}$", abs(-6:-1)))
  )+
  ylim(-100,0)+
  theme_minimal(base_family = "Times New Roman")
# myplot

if(SAVEPLOTS){
  ggsave(filename = "figs/scenario1b_pi_nonoise.pdf",
            plot = myplot,
            height = 4,width = 4.4 
            )
    # ggsave(filename = "figs/scenario1b_pi_nonoise-lotsoflabels.pdf",
    #         plot = myplot,
    #         height = 4,width = 4.4 
            # )
}

replicates=100

# Wrapper for conciseness
datsim<-wrapper_model(dfpicisubset,replicates,"Pop.Bottleneck",
                      G_inv_drift_population_loss, round(dfpicisubset$Gen.interval)) %>%
  mutate(Gen.interval=rep(dfpicisubset$Gen.interval,replicates)) %>%
  arrange(abs(pvalue_color))

# plot
myplot2<-
ggplot(datsim)+
  geom_point(aes(y=Percent.Change.original,
                 x=-log10(Pop.Bottleneck),
                 size=Gen.interval,
                 color=factor(pvalue_color)
  ),
  # height = 1, width = 1,
  alpha=0.5,shape=16,shape=16)+
  scale_color_manual("Trend", values = c(
    "red",
    "grey",
    "navy"
  ))+
  xlab(TeX("Population with drift (N)"))+
  ylab("Percent diversity change (π%)")+
  scale_size_continuous("Generations",range = c(0.5,4), breaks = c(1,25,50,75))+
  scale_x_continuous(
    breaks = -6:-1,
    limits = c(-6, -1),
    labels = TeX(sprintf("$10^{%d}$", abs(-6:-1)))
  )+
  ylim(-100,0)+
  theme_minimal(base_family = "Times New Roman")
# myplot2

if(SAVEPLOTS){
  ggsave(filename = "figs/scenario1b_pi_noise.pdf",
            plot = myplot2,
            height = 4,width = 4.4
            )
}

# PLOT GRID
# plot_grid(myplot+theme(legend.position = "none"),myplot2,rel_widths = c(0.8,1))
plot_grid(myplot,myplot2)

```

\
NB. Positive trend % of genetic diversity collapsed to 0%.\
Number of generations truncated to 100+ for visualization purposes.\
Number of population size needed to explain loss truncated to 1M.\

*No noise*\
Median = `r printmedian(res$Pop.Bottleneck[res$pvalue_color== -1])` individuals.\
Mean = `r printmean(res$Pop.Bottleneck[res$pvalue_color== -1])` individuals.\
IQR = `r printIQR(res$Pop.Bottleneck[res$pvalue_color== -1])` individuals.\
Ranges = `r printranges(res$Pop.Bottleneck[res$pvalue_color== -1])` individuals.\

*Noise*\
Median = `r printmedian(datsim$Pop.Bottleneck[datsim$pvalue_color==-1])`individuals.\
Mean = `r printmean(datsim$Pop.Bottleneck[datsim$pvalue_color==-1])`individuals.\
IQR = `r printIQR(datsim$Pop.Bottleneck[datsim$pvalue_color==-1])`individuals.\
Ranges = `r printranges(datsim$Pop.Bottleneck[datsim$pvalue_color==-1])`individuals.\

```{r, eval=F, echo=F, message=F, error=F}

dplyr::filter(res, Binomen.rotl=="Strigops habroptila")$Percent.Change
dplyr::filter(res, Binomen.rotl=="Strigops habroptila")$Pop.Bottleneck
dplyr::filter(d, Binomen.rotl=="Strigops habroptila")$Citation


```

#### *Allelic richnes* $M$

```{r fig:drift_m_noise ,echo=FALSE, warning=FALSE, eval=T,fig.width=7,fig.height=3.5, fig.cap="Expected immediate population bottleneck given observed M%. \\label{fig:Drift_M_panmictic_noise_grid}"}

mininferred=1e6
# run inverse modeling
N0=1e6
res=M_inv_drift_population_loss(dfmcisubset, N0)

# add and round intervals to the results
res$Gen.interval<-round(dfmsubset$Gen.interval+1)
res$Gen.interval[res$Gen.interval>100]<-100

# add the hedge g p vlauem and arrange by p value
res$pvalue_color<-dfmcisubset$pvalue_color
res<-res %>% arrange(abs(pvalue_color))

# limit the higher side of population bottleneck
res$Pop.Bottleneck[is.infinite(res$Pop.Bottleneck)]<- mininferred
res$Pop.Bottleneck[res$Pop.Bottleneck>mininferred]<-mininferred

# plot
myplot<-
ggplot(res)+
  geom_point(aes(y=Percent.Change,
                 x=-log10(Pop.Bottleneck),
                 size=Gen.interval,
                 color=factor(pvalue_color)
  ),
  # height = 1, width = 1,
  alpha=0.5,shape=16,shape=16)+
  scale_color_manual("Trend", values = c(
    "red",
    "grey",
    "navy"
  ))+
  xlab(TeX("Population with drift (N)"))+
  ylab("Percent diversity change (M%)")+
  scale_size_continuous("Generations",range = c(0.5,4), breaks = c(1,25,50,75))+
  scale_x_continuous(
    breaks = -6:-0,
    limits = c(-6, -0),
    labels = TeX(sprintf("$10^{%d}$", abs(-6:-0)))
  )+
  ylim(-100,0)+
  theme_minimal(base_family = "Times New Roman")

# myplot

if(SAVEPLOTS){
  ggsave(filename = "figs/scenario1b_m_nonoise.pdf",
            plot = myplot,
            height = 4,width = 4.4
            )
}

replicates=100
N0=1e6

# Wrapper for conciseness
datsim<-wrapper_model(dfmcisubset,replicates,relevantvariable="Pop.Bottleneck",
                      M_inv_drift_population_loss, N0=N0 ) %>%
  mutate(Gen.interval=rep(dfmcisubset$Gen.interval,replicates)) %>%
  arrange(abs(pvalue_color))

# Plot
myplot2<-
ggplot(datsim)+
  geom_point(aes(y=Percent.Change.original,
                 x=-log10(Pop.Bottleneck),
                 size=Gen.interval,
                 color=factor(pvalue_color)
  ),
  # height = 1, width = 1,
  alpha=0.5,shape=16,shape=16)+
  scale_color_manual("Trend", values = c(
    "red",
    "grey",
    "navy"
  ))+
  xlab(TeX("Population with drift (N)"))+
  ylab("Percent diversity change (M%)")+
  scale_size_continuous("Generations",range = c(0.5,4), breaks = c(1,25,50,75))+
  scale_x_continuous(
    breaks = -6:-0,
    limits = c(-6, -0),
    labels = TeX(sprintf("$10^{%d}$", abs(-6:-0)))
  )+
  ylim(-100,0)+
  theme_minimal(base_family = "Times New Roman")

if(SAVEPLOTS){
  ggsave(filename = "figs/scenario1b_m_noise.pdf",
            plot = myplot2,
            height = 4,width = 4.4
            )
}

plot_grid(myplot,myplot2)

```

Population bottleneck truncated up to $10^5$ for visualization\
Positive trend % of genetic diversity collapsed to 0%.\
Starting population also needs to be assumed, $N_0=10^6$, which is a conservative value (as a smaller starting opulation is assumed, the bottleenck predicted is even larger)

Expected instantaneous population bottleneck percentage for studies with significan decline of $M$ :\
Median = `r printmedian((datsim$Pop.Bottleneck[datsim$pvalue_color==-1]))`% of original population\
IQR = `r printIQR((datsim$Pop.Bottleneck[datsim$pvalue_color==-1]))`\

\clearpage

### SPACE. Spatial meta-populations, immediate -- *scenario 2a*

#### *Average genetic diversity* $\pi$

```{r create_GDAR ,echo=FALSE, warning=FALSE,fig.width=4,fig.height=3.5, fig.cap="Expected immediate geographic area loss given observed π%. "}

res=G_inv_immediate_area_loss_GDAR(dfpicisubset)

res$pvalue_color<-dfpicisubset$pvalue_color
res$Binomen.rotl<-dfpicisubset$Binomen.rotl

res<-res %>% arrange(abs(pvalue_color))

# plot
myplot<-
ggplot(res)+
  geom_point(aes(y=Percent.Change,
                  x=Area.Loss,
                  color=factor(pvalue_color)),
              height = 1, width = 1, alpha=0.5,shape=16)+
    scale_color_manual("Trend", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  ggrepel::geom_text_repel(
    data = dplyr::filter(res,pvalue_color==-1),
    aes(y=Percent.Change,
        x=Area.Loss,
        color=factor(pvalue_color), 
        label = Binomen.rotl),
   family = "Times New Roman",
      fontface = "italic",
      max.overlaps = 200,
      force = 0.1,           # increase repelling strength
      force_pull = 0.01,      # keep connection to origin
      nudge_x = 0.2         # nudge outward (you can try nudge_y = 2 instead or as well)
  ) +
  xlab("Area loss (%)")+
  ylab("Percent diversity change (π%)")+
  xlim(0,120)+ylim(-100,0)+
  theme_minimal(base_family = "Times New Roman")
  
ggsave(filename = "figs/scenario2a_pi_nonoise-lotslabels.pdf",
            plot = myplot,
            height = 20,width = 20
            )

  myplot<-
ggplot(res)+
  geom_jitter(aes(y=Percent.Change,
                  x=Area.Loss,
                  color=factor(pvalue_color)),
              height = 1, width = 1, alpha=0.5,shape=16)+
    scale_color_manual("Trend", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  xlab("Area loss (%)")+
  ylab("Percent diversity change (π%)")+
  xlim(0,100)+ylim(-100,0)+
  theme_minimal(base_family = "Times New Roman")
    ggsave(filename = "figs/scenario2a_pi_nonoise-lotslabels.pdf",
            plot = myplot,
            height = 20,width = 20
            )

# myplot

if(SAVEPLOTS){
  ggsave(filename = "figs/scenario2a_pi_nonoise.pdf",
            plot = myplot,
            height = 4,width = 4
            )
}


# WITH NOISE

replicates=100

# Make an original copy
datori<-dfpicisubset

# Reample dataset based on noise
datsim<-resample(dfpicisubset,replicates)
datsim$Percent.Change[is.na(datsim$Percent.Change)]<-0

# Inverse modeling
resori<-M_inv_immediate_area_loss_MAR(datori)
ressim=M_inv_immediate_area_loss_MAR(datsim)

# Put together dataset with extra info on trends
datori$Area.Loss<-resori$Area.Loss
datori$pvalue_color<-dfpicisubset$pvalue_color
datsim$Area.Loss<-ressim$Area.Loss
datsim$pvalue_color<-rep(dfpicisubset$pvalue_color,replicates) # nolte this will keep filling

# Add the original to the simulated
datsim$Area.Loss.original<-rep(resori$Area.Loss,replicates)
datsim$Percent.Change.original<-rep(resori$Percent.Change,replicates)

# Plot the
datsim<-datsim %>% arrange(abs(pvalue_color))
myplot2<-
ggplot(datsim)+
  geom_point(aes(y=Percent.Change.original,
                 x=Area.Loss,
                color=factor(pvalue_color)),
           height = 1, width = 1, alpha=0.5,shape=16)+
  scale_color_manual("Trend", values = c(
    "red",
    "grey",
    "navy"
  ))+
  xlab("Area loss (%)")+
  ylab("Percent diversity change (π%)")+
  xlim(0,100)+ylim(-100,0)+
  theme_minimal(base_family = "Times New Roman")

# myplot2

if(SAVEPLOTS){
  ggsave(filename = "figs/scenario2a_pi_noise.pdf",
            plot = myplot2,
            height = 4,width = 4
            )
}

```

```{r fig:GDAR ,echo=FALSE, warning=FALSE,fig.width=7,fig.height=3.5, fig.cap="Expected immediate geographic area loss given observed π%.  \\label{fig:GDAR_panmictic_noise_grid}"}

plot_grid(myplot, myplot2)

```

NB. A jitter of 1% area units was added for visualization. Positive trend % of genetic diversity collapsed to 0%.\

*No noise*\
Inverse modeling of instantaneous $M$ richness loss with spatial contraction:\
Median = `r printmedian(res$Area.Loss[res$pvalue_color==-1])`% area\
Mean = `r printmean(res$Area.Loss[res$pvalue_color==-1])`% area\
IQR = `r printIQR(res$Area.Loss[res$pvalue_color==-1])`% area\
Range = `r printranges(res$Area.Loss[res$pvalue_color==-1])`% area\

*Noise*\
Inverse modeling of instantaneous $M$ richness loss with spatial contraction:\
Median = `r printmedian((datsim$Area.Loss[datsim$pvalue_color==-1]))`% area\
Mean = `r printmean((datsim$Area.Loss[datsim$pvalue_color==-1]))`% area\
IQR = `r printIQR((datsim$Area.Loss[datsim$pvalue_color==-1]))`% area\
Range = `r printranges((datsim$Area.Loss[datsim$pvalue_color==-1]))`% area\

Expected instantaneous area contraction percentage for studies with significan decline of $\pi$ :\
Median = `r printmedian((datsim$Area.Loss[datsim$pvalue_color==-1]))`% of original range\
IQR = `r printIQR((datsim$Area.Loss[datsim$pvalue_color==-1]))`\

```{r echo=F, eval=FALSE}

dplyr::filter(res,Area.Loss >90)$Binomen.rotl

dplyr::filter(res,Binomen.rotl == "Ursus thibetanus")

```

#### *Allelic richnes* $M$

```{r fig:MAR_noise ,echo=FALSE, warning=FALSE,fig.width=7,fig.height=3.5, fig.cap="Expected immediate geographic area loss given observed M%.  \\label{fig:MAR_panmictic_noise_grid}"}


res=M_inv_immediate_area_loss_MAR(dfmcisubset)

res$pvalue_color<-dfmcisubset$pvalue_color

res<-res %>% arrange(abs(pvalue_color))

# plot
myplot<-
ggplot(res)+
  geom_jitter(aes(y=Percent.Change,
                  x=Area.Loss,
                  color=factor(pvalue_color)),
              height = 1, width = 1, alpha=0.5,shape=16)+
    scale_color_manual("Trend", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  xlab("Area loss (%)")+
  ylab("Percent diversity change (M%)")+
  xlim(0,100)+ylim(-100,0)+
  theme_minimal(base_family = "Times New Roman")
# print(myplot)

if(SAVEPLOTS){
  ggsave(filename = "figs/scenario2b_pi_nonoise.pdf",
            plot = myplot,
            height = 4,width = 4
            )
}


replicates=100

# Make an original copy
datori<-dfmci

# Reample dataset based on noise
datsim<-resample(dfmci,replicates)
datsim$Percent.Change[is.na(datsim$Percent.Change)]<-0

# Inverse modeling
resori<-M_inv_immediate_area_loss_MAR(datori)
res=M_inv_immediate_area_loss_MAR(datsim)

# Put together dataset with extra info on trends
datori$Area.Loss<-resori$Area.Loss
datori$pvalue_color<-dfmci$pvalue_color
datsim$Area.Loss<-res$Area.Loss
datsim$pvalue_color<-rep(dfmci$pvalue_color,replicates) # nolte this will keep filling

# Add the original to the simulated
datsim$Area.Loss.original<-rep(resori$Area.Loss,replicates)
datsim$Percent.Change.original<-rep(resori$Percent.Change,replicates)

# Plot the
datsim<-datsim %>% arrange(abs(pvalue_color))
myplot2<-
ggplot(datsim)+
  geom_point(aes(y=Percent.Change.original,
                 x=Area.Loss,
                color=factor(pvalue_color)),
           height = 1, width = 1, alpha=0.5,shape=16)+
  scale_color_manual("Trend", values = c(
    "red",
    "grey",
    "navy"
  ))+
  xlab("Area loss (%)")+
  ylab("Percent diversity change (M%)")+
  xlim(0,100)+ylim(-100,0)+
  theme_minimal(base_family = "Times New Roman")


# myplot2

if(SAVEPLOTS){
  ggsave(filename = "figs/scenario2b_m_noise.pdf",
            plot = myplot2,
            height = 4,width = 4
            )
}


plot_grid(myplot, myplot2)

```

Positive trend % of genetic diversity collapsed to 0%.\

The above reconstructions take at face value the average genetic diversity change, but as seen in Text S2 there is much noise in these estimates. A more true visualization would re-generate each observation 100 times following the observed error in the quantity.

Expected instantaneous area contraction percentage for studies with significan decline of $M$ :\
Median = `r printmedian((datsim$Area.Loss[datsim$pvalue_color==-1]))`% of original range.\
IQR = `r printIQR((datsim$Area.Loss[datsim$pvalue_color==-1]))`.\

\clearpage

### SPACE & TIME. Spatial meta-populations via WFmoments, short drift -- *scenario 2b*

Both spatial contraction and short drift. This is the most complicated scenario as we do not have a direct simple mathematical method. Instead, we have pre-computed scenarios of a solved numerical method we call WFmoments, that can predict the non-equilibrium genetic diversity of any meta-population system over time. We also only have solutions for average genetic diversity π. In this approach, then we created a number of loss area scenarios in a system of 100 demes in a 2D lattice, losing demes in an edge contraction manner. We then inverse model the area loss corresponding to the matched π change simulated and the observed π changes. We can use multiple functions, although they give qualitatively similar results. We decide for a Gaussian kernel, where the distance between the observed and simulated π changes exponentially decreases.

By definition, meta-population dynamics will be most relevant to species with multiple populations, and thus may not be as affected by single population dynamics (i.e. with a number of populations in the landscape, their effective population size will unlikely be below the \~500 where we see single population drift losses). Then, when in a meta-population landscape some populations are lost, drift will increase but most importantly the whole system will tend towards a future lower genetic diversity equilibrium. Population genetics tells us that for populations to reach equilibrium about 4Ne generations. Assuming different starting $N_0$ values and using the information of time interval, we can then compute

#### *Average genetic diversity* $\pi$

```{r WF model , echo=FALSE, warning=FALSE,error=FALSE,message=FALSE}

library(data.table)
# wf <- fread("~/Shareddrives/MOI-LAB/PROJECTS/extinction/MAR2.0/Data policy/WFmoments_EDGE_RedList.tsv.gz")

edgescenarios<-"WFmoments_approximation_table_edge_contraction.tsv"
fragmentationscenarios<-"WFmoments_approximation_table_fragmentation.tsv"

wf <- wffrag <- fread(fragmentationscenarios) %>%
  mutate(Percent.Change = `% short_term_per-deme_pi` -100) %>%
  mutate(Percent.Change.Fut = `% long_term_per-deme_pi` -100) %>%
  mutate(Area.Loss=`% habitat loss`*100)
wf <- wfedge <- fread(edgescenarios) %>%
  mutate(Percent.Change = `% short_term_species_wide_pi` -100) %>%
  mutate(Percent.Change.Fut = `% long_term_species_wide_pi` -100) %>%
  mutate(Area.Loss=`% habitat loss`*100) %>%
  dplyr::filter(avg_fst_nei<0.5)


# add the rows
add0<-data.frame(matrix(ncol = ncol(wf), nrow = table(wfedge$Area.Loss)[1]))
colnames(add0)<-colnames(wfedge)
add0$Percent.Change<- -100
add0$Area.Loss<- 100

# add -100, this could be problematic
# wfedge<-rbind(wfedge,add0)
# wffrag<-rbind(wffrag,add0)

```

```{r check_simulations, echo=F, warning=F,eval=F}

wffrag$Area.Loss %>% summary
wfedge$Area.Loss %>% summary

qplot(y=wffrag$Percent.Change,x=wffrag$Area.Loss) #+xlim(0,100)
qplot(y=wfedge$Percent.Change,x=wfedge$Area.Loss) #+xlim(0,100)

```

```{r check_simulations2, echo=F, warning=F,eval=F}

wfedge$Percent.Change %>% round %>% unique

wfedge %>%
  mutate(Percent.Change.round= round(Percent.Change)) %>%
  group_by(Percent.Change.round) %>%
  summarise(meanAx= mean(Area.Loss),
            medianAx= mean(Area.Loss))


```

```{r WFmoments ,echo=FALSE, warning=FALSE, echo=FALSE, warning=FALSE,fig.width=4,fig.height=3.5}

res=G_inv_drift_area_loss_WFmoments_sample(dfpicisubset, wfedge, weighting_function_Gaussian)

# color p values
res$pvalue_color<-dfpicisubset$pvalue_color
res<-res %>% arrange(abs(pvalue_color))
# Reassign zero to the positive trends (as we use)
res<-res %>% arrange(abs(pvalue_color))

# plot
myplot<-
  ggplot(res)+
  geom_jitter(aes(y=Percent.Change,
                  x=Area.Loss,
                  color=factor(pvalue_color)),
              height = 1, width = 1, alpha=0.5,shape=16)+
  scale_color_manual("Trend", values = c(
    "red",
    "grey",
    "navy"
  ))+
  xlab("Area loss (%)")+
  ylab("Percent diversity change (M%)")+
  xlim(0,100)+ylim(-100,0)+
  theme_minimal(base_family = "Times New Roman")
# print(myplot)


if(SAVEPLOTS){
  ggsave(filename = "figs/scenario2b_pi_nonoise.pdf",
         plot = myplot,
         height = 4,width = 4
  )
}

```

```{r fig:WFmoments_time_noise ,echo=FALSE, warning=FALSE,fig.width=7,fig.height=3.5, fig.cap="Expected immediate geographic area loss given observed π%. \\label{fig:WFmoments}"}

res=G_inv_drift_area_loss_WFmoments_drift_sample(dfpicisubset,
                                           Nt=100,
                                           wfedge,
                                           weighting_function_Gaussian)

# add generations
res$Gen.interval=dfpicisubset$Gen.interval
# color p values
res$pvalue_color<-dfpicisubset$pvalue_color
res<-res %>% arrange(abs(pvalue_color))
# Reassign zero to the positive trends (as we use)
res<-res %>% arrange(abs(pvalue_color))

# plot
myplot<-
  ggplot(res)+
  geom_jitter(aes(y=Percent.Change,
                  size=Gen.interval,
                  x=Area.Loss,
                  color=factor(pvalue_color)),
              height = 1, width = 1, alpha=0.5,shape=16)+
  scale_color_manual("Trend", values = c(
    "red",
    "grey",
    "navy"
  ))+
  scale_size_continuous("Generations",range = c(0.5,5.5), breaks = c(1,25,50,75))+

  xlab("Area loss and drift (%)")+
  ylab("Percent diversity change (π%)")+
  xlim(0,100)+ylim(-100,0)+
  theme_minimal(base_family = "Times New Roman")

# print(myplot)


# if(SAVEPLOTS){
  ggsave(filename = "figs/scenario2b_time_pi_nonoise.pdf",
         plot = myplot,
         height = 4,width = 4
  )


#####*********************************************************************######
replicates=100
N1=1e3

redo=TRUE


if(redo){
datori<-dfpicisubset

# Reample dataset based on noise
datsim<-resample(datori,replicates)
datsim$Percent.Change[is.na(datsim$Percent.Change)]<-0

# Inference
resori=G_inv_drift_area_loss_WFmoments_drift_sample(datori,
                                         Nt=100,
                                         wfedge,
                                         weighting_function_Gaussian)
res=G_inv_drift_area_loss_WFmoments_drift_sample(datsim,
                                         Nt=100,
                                         wfedge,
                                         weighting_function_Gaussian)
# Put together dataset with extra info on trends
datori$Area.Loss<-resori$Area.Loss
datori$pvalue_color<-datori$pvalue_color
datsim$Area.Loss<-res$Area.Loss
datsim$pvalue_color<-rep(datori$pvalue_color,replicates) # nolte this will keep filling

# Add the original to the simulated
datsim$Area.Loss.original<-rep(datori$Area.Loss,replicates)
datsim$Percent.Change.original<-rep(datori$Percent.Change,replicates)

save(file = "tmp/datsim_wfmoments_scenario2b.rda", datsim)

}else{
  load("tmp/datsim_wfmoments_scenario2b.rda")
}


# Sort
datsim<-
  datsim %>%
  arrange(abs(pvalue_color))

# Plot
myplot2<-
ggplot(datsim)+
  geom_jitter(aes(y=Percent.Change.original,
                  size=Gen.interval,
                  x=Area.Loss,
                  color=factor(pvalue_color)),
              height = 1, width = 1, alpha=0.5,shape=16)+
  scale_color_manual("Trend", values = c(
    "red",
    "grey",
    "navy"
  ))+
  scale_size_continuous("Generations",range = c(0.5,5.5), breaks = c(1,25,50,75))+

  xlab("Area loss and drift (%)")+
  ylab("Percent diversity change (π%)")+
  xlim(0,100)+ylim(-100,0)+
  theme_minimal(base_family = "Times New Roman")


# GRID
plot_grid(myplot, myplot2)

```

Summary:

*No noise*\
Median = `r printmedian(res$Area.Loss[res$pvalue_color==-1])`% of original range lost.\
Mean = `r printmean(res$Area.Loss[res$pvalue_color==-1])`% of original range lost.\
IQR = `r printIQR(res$Area.Loss[res$pvalue_color==-1])`% of original range lost.\
Ranges = `r printranges(res$Area.Loss[res$pvalue_color==-1])`% of original range lost.\

*Noise*\
Median = `r printmedian(datsim$Area.Loss[datsim$pvalue_color==-1])`% of original range lost.\
Mean = `r printmean(datsim$Area.Loss[datsim$pvalue_color==-1])`% of original range lost.\
IQR = `r printIQR(datsim$Area.Loss[datsim$pvalue_color==-1])`% of original range lost.\
Ranges = `r printranges(datsim$Area.Loss[datsim$pvalue_color==-1])`% of original range lost.\

#### *Allelic richness* $M$

We have no model for this.

\clearpage

## Text S5: Interpret increase in genetic diversity

There are three options to interpret population genetic diversity: (1) that genetic diversity is indeed increasing species-wide, (2) genetic diversity is inflated due to changes in meta-population dynamics, (3) measurement error.

### SINGLE POPULATION GROWTH OVER TIME -- *gain scenario 1b*

```{r fig:popgrowth, echo=FALSE, warning=FALSE, echo=FALSE, warning=FALSE,fig.width=10,fig.height=3.5, fig.cap="Expected population growth given increases in π%. \\label{fig:increases}"}


N1=100
# Inverse modeling
res=G_inv_growth(dfpicisubset,
                 N1,
                 t = round(dfpisubset$Gen.interval+1))
  res$Gen.interval<-round(dfpisubset$Gen.interval+1)
  res$Gen.interval[res$Gen.interval>100]<-100
  res$pvalue_color<-dfpicisubset$pvalue_color
  res<-res %>% arrange(abs(pvalue_color))

# plot
plota<-
ggplot(res)+
  geom_point(aes(y=Percent.Change,
                 x=log10(Pop.Growth),
                 size=Gen.interval,
                 color=factor(pvalue_color)
  ),
  # height = 1, width = 1,
  alpha=0.5,shape=16,shape=16)+
  scale_color_manual("Trend", values = c(
    "red",
    "grey",
    "navy"
  ))+
  xlab(TeX("Population growth (N%)"))+
  ylab("Percent diversity change (π%)")+
  scale_size_continuous("Generations",range = c(0.5,4), breaks = c(1,25,50,75))+
  scale_x_continuous(
    breaks = c(0:8),
    limits = c(0,8),
    labels = TeX(sprintf("$10^{%d}$", 0:8))
  )+
  geom_vline(xintercept = 2, lty="dotted")+ #100% marked
  theme_minimal(base_family = "Times New Roman")


N1=1000
# Inverse modeling
res=G_inv_growth(dfpicisubset,
                 N1,
                 t = round(dfpisubset$Gen.interval+1))
  res$Gen.interval<-round(dfpisubset$Gen.interval+1)
  res$Gen.interval[res$Gen.interval>100]<-100
  res$pvalue_color<-dfpicisubset$pvalue_color
  res<-res %>% arrange(abs(pvalue_color))

# plot
plotb<-
ggplot(res)+
  geom_point(aes(y=Percent.Change,
                 x=log10(Pop.Growth),
                 size=Gen.interval,
                 color=factor(pvalue_color)
  ),
  # height = 1, width = 1,
  alpha=0.5,shape=16,shape=16)+
  scale_color_manual("Trend", values = c(
    "red",
    "grey",
    "navy"
  ))+
  xlab(TeX("Population growth (N%)"))+
  ylab("Percent diversity change (π%)")+
  scale_size_continuous("Generations",range = c(0.5,4), breaks = c(1,25,50,75))+
  scale_x_continuous(
    breaks = c(0:8),
    limits = c(0,8),
    labels = TeX(sprintf("$10^{%d}$", 0:8))
  )+
  geom_vline(xintercept = 2, lty="dotted")+ #100% marked
  theme_minimal(base_family = "Times New Roman")

N1example<-N1
resexample<-res


N1=10000
# Inverse modeling
res=G_inv_growth(dfpicisubset,
                 N1,
                 t = round(dfpisubset$Gen.interval+1))
  res$Gen.interval<-round(dfpisubset$Gen.interval+1)
  res$Gen.interval[res$Gen.interval>100]<-100
  res$pvalue_color<-dfpicisubset$pvalue_color
  res<-res %>% arrange(abs(pvalue_color))


# plot
plotc<-
ggplot(res)+
  geom_point(aes(y=Percent.Change,
                 x=log10(Pop.Growth),
                 size=Gen.interval,
                 color=factor(pvalue_color)
  ),
  # height = 1, width = 1,
  alpha=0.5,shape=16,shape=16)+
  scale_color_manual("Trend", values = c(
    "red",
    "grey",
    "navy"
  ))+
  xlab(TeX("Population growth (N%)"))+
  ylab("Percent diversity change (π%)")+
  scale_size_continuous("Generations",range = c(0.5,4), breaks = c(1,25,50,75))+
  scale_x_continuous(
    breaks = c(0:8),
    limits = c(0,8),
    labels = TeX(sprintf("$10^{%d}$", 0:8))
  )+
  geom_vline(xintercept = 2, lty="dotted")+ #100% marked
  theme_minimal(base_family = "Times New Roman")



N1=1e6
# Inverse modeling
res=G_inv_growth(dfpicisubset,
                 N1,
                 t = round(dfpisubset$Gen.interval+1))
  res$Gen.interval<-round(dfpisubset$Gen.interval+1)
  res$Gen.interval[res$Gen.interval>100]<-100
  res$pvalue_color<-dfpicisubset$pvalue_color
  res<-res %>% arrange(abs(pvalue_color))

# plot
plotd<-
ggplot(res)+
  geom_point(aes(y=Percent.Change,
                 x=log10(Pop.Growth),
                 size=Gen.interval,
                 color=factor(pvalue_color)
  ),
  # height = 1, width = 1,
  alpha=0.5,shape=16,shape=16)+
  scale_color_manual("Trend", values = c(
    "red",
    "grey",
    "navy"
  ))+
  xlab(TeX("Population growth (N%)"))+
  ylab("Percent diversity change (π%)")+
  scale_size_continuous("Generations",range = c(0.5,4), breaks = c(1,25,50,75))+
  scale_x_continuous(
    breaks = c(0:8),
    limits = c(0,8),
    labels = TeX(sprintf("$10^{%d}$", 0:8))
  )+
  geom_vline(xintercept = 2, lty="dotted")+ #100% marked
  theme_minimal(base_family = "Times New Roman")


plot_grid(plota+theme(legend.position = "none"),
          plotb+theme(legend.position = "none"),
          plotc+theme(legend.position = "none"),
          plotd+theme(legend.position = "none"), nrow=1)


```

Fig. \ref{fig:fst} shows plots assuming final population sizes: $N1=100, 1000, 10000 , 1000000$. All data points of decline area assumed to be 0%. Dotted line marks the 100% increase, or doubling of population.

Fig. \ref{fig:increasedecrease} explains why growth of genetic diversity is even less expected than declines. It assumes a population started at 10,000 individuals, and went down to 1,000 or increased to 15,000.

```{r fig:increasedecrease, echo=FALSE, warning=FALSE, echo=FALSE, warning=FALSE,fig.width=3,fig.height=2.5, fig.cap="The shape of genetic diversity increases is slower than decreases. \\label{fig:increasedecrease}"}

N0        <- 1000                 # ancestral effective size
scenarios <- tibble(
  scenario = c("Decline: 1000 → 500", "Growth: 1000 → 1500"),
  N1       = c(500, 1500)
)
t_max <- 4 *N0*2                     

pi_frac <- function(N1, t, N0) {
  r <- N1 / N0                               # N1 / N0
  r + (1 - r) * (1 - 1 / (2 * N1))^t         # π_t / π_0
}

plot_df <- expand_grid(t = 0:t_max, scenarios) %>%
  mutate(pi_over_pi0 = pi_frac(N1, t, N0))

ggplot(plot_df, aes(x = t, y = (pi_over_pi0-1)*100, colour = scenario)) +
  geom_line(size = 1) +
  labs(
    x       = "Generations since size change",
    y       = "Percent change diversity (%π)",
    # y       = expression(pi[t] / pi[0]),
    colour  = "Scenario"#,
  ) +
  theme_minimal(base_family = "Times New Roman")+
  scale_color_manual(values = c("red","navy"))+

  theme(
    legend.position = "top",
    legend.title    = element_text(size = 10),
    legend.text     = element_text(size = 9)
  )

```

\clearpage

### SPATIAL METAPOPULATIONS MIXING -- *gain scenario 2a*

```{r fig:FST ,echo=FALSE, warning=FALSE, echo=FALSE, warning=FALSE,fig.width=4,fig.height=3.5, fig.cap="Expected $F_{ST}$ to cause a inflation of π%.\\label{fig:fst} "}

res<-G_inv_Fst(dfpicisubset)
res$pvalue_color<-dfpicisubset$pvalue_color
# res<-res %>% arrange(abs(pvalue_color))

fstplot<-
ggplot(res)+
  geom_jitter(aes(y=Percent.Change,
                  x=Fst,
                  color=factor(pvalue_color)),
              height = 1, width = 0.01,
             alpha=0.5,shape=16)+
    scale_color_manual("Trend", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  xlab(TeX("$F_{ST}$"))+
  ylab("Percent diversity change (π%)")+
  xlim(-0.05,0.5)+
  # ylim(0,100)+
  theme_minimal(base_family = "Times New Roman")

fstplot

SAVEPLOTS=F
if(SAVEPLOTS){
  ggsave(filename = "figs/fstplot.pdf",
            plot = fstplot,
            height = 4,width = 4
            )
}

```

The median $F_{ST}$ required for a mixing of two popualtions to cause the percentage genetic change increase observed in positive cases will be: `r median(res$Fst[res$pvalue_color==1])` (\ref{fig:fst}).

```{r fig:WFmomentsfragmentation ,echo=FALSE, warning=FALSE, echo=FALSE, warning=FALSE,fig.width=4,fig.height=3.5, fig.cap="Expected noisy inflation of π% from area loss and fragmentation.\\label{fig:fragmentation} "}
# Need to reload to take the species wide fragmentation scenario

wf <- wffrag <- fread(fragmentationscenarios) %>%
  # mutate(Percent.Change = `% short_term_per-deme_pi` -100) %>%  # choose speices?
  mutate(Percent.Change = `% short_term_species_wide_pi` -100) %>%
  mutate(Percent.Change.Fut =`% long_term_species_wide_pi` -100) %>%
  # take the  species wide that could be the cause of inflation
  mutate(Area.Loss=`% habitat loss`*100) %>%
  filter(avg_fst_nei<0.5)


# Inverse modeling
dfpicitmp<-
  dfpicisubset %>%
  mutate(Percent.Change=ifelse(Percent.Change<0, 0,Percent.Change)) # ensure no positive



# this takes generations into account
res=G_inv_drift_area_loss_WFmoments_drift_sample(dat=dfpicitmp,
                                                 Nt=1000,
                                                 wfmoments=wffrag
)
res$pvalue_color<-dfpicisubset$pvalue_color
res<-res %>% arrange(abs(pvalue_color))

resfragmentation<-res

# plot
myplot<-
ggplot(res)+
  geom_point(aes(y=Percent.Change,
                  x=Area.Loss,
                  color=factor(pvalue_color)),
              height = 1, width = 1, alpha=0.5,shape=16)+
    scale_color_manual("Trend", values = c(
                                                "red",
                                                "grey",
                                                "navy"
                                                ))+
  xlab("Area loss and fragmentation (%)")+
  ylab("Percent diversity change (π%)")+
  xlim(0,100)+
  ylim(0,100)+
  theme_minimal(base_family = "Times New Roman")

 print(myplot)


if(SAVEPLOTS){
  ggsave(filename = "figs/scenario2b_pi_nonoise_fragmentation.pdf",
            plot = myplot,
            height = 4,width = 4
            )
}

```

NB \ref{fig:fragmentation}. Plot truncated to up to +100% in genetic diversity. Negative diversity trends truncated.\

This also shows that it could be that even if genetic diversity increases, area loss and fragmentation could have happened.

Expected instantaneous area contraction percentage for studies with significan decline of $\pi$ :\
Mean = `r printmean((res$Area.Loss[res$pvalue_color==1]))`% of original range.\
Median = `r printmedian((res$Area.Loss[res$pvalue_color==1]))`% of original range.\
IQR = `r printIQR((res$Area.Loss[res$pvalue_color==1]))`% of original range.\
Range = `r printranges((res$Area.Loss[res$pvalue_color==1]))`% of original range.\

<!-- ## Text S6 Possible future scenarios -->

## Text S6: Simple population contraction averages from LPI

```{r lpi, echo=FALSE, warning=FALSE, fig.width=4, fig.height=3.5, fig.cap="Res loss from geographic area reduction on π%.\\label{fig:fragmentation}"}

REDO = F

if (REDO) {
  file_path <- "../dataLPI/LPD_2024_public.csv"
  d <- read.csv(file_path, header = TRUE)

  # Estimate the trajectory of population using the log-linear pop model
  get_lm_slope <- function(pop, years) {
    valid <- !is.na(pop)
    if (sum(valid) >= 2) {
      model <- lm(log10(pop[valid] + 1) ~ years[valid])
      return(coef(model)[2])  # return the slope
    } else {
      return(NA_real_)
    }
  }

  percent_change_from_log_lambda <- function(log_lambda, t) {
    if (!is.na(log_lambda) && !is.na(t)) {
      return((exp(t * log_lambda) - 1) * 100)
    } else {
      return(NA_real_)
    }
  }

  dsum <- d %>%
    rowwise() %>%
    mutate(
      across(starts_with("X"), ~ as.numeric(na_if(., "NULL"))),
      pop_vector = list(c_across(X1967:X2020)),
      year_vector = list(1967:2020),
      lm_slope = get_lm_slope(pop_vector, year_vector)
    ) %>%
    mutate(
      minyear = if (!all(is.na(pop_vector))) min(year_vector[!is.na(pop_vector)]) else NA_integer_,
      maxyear = if (!all(is.na(pop_vector))) max(year_vector[!is.na(pop_vector)]) else NA_integer_,
      minyearN = if (!all(is.na(pop_vector))) pop_vector[which(year_vector == minyear)] else NA_integer_,
      maxyearN = if (!all(is.na(pop_vector))) pop_vector[which(year_vector == maxyear)] else NA_integer_
    ) %>%
    mutate(
      diffabundance = (maxyearN / minyearN) - 1,
      diffyears = maxyear - minyear,
      percent_change = percent_change_from_log_lambda(lm_slope, diffyears)
    ) %>%
    ungroup()

  # Summarize by species using simple percentage abundance change
  summary_df_species_simple <- dsum %>%
    group_by(Binomial) %>%
    summarise(
      mean_slope= mean(lm_slope, na.rm=T),
      mean_abundance_change = mean(diffabundance, na.rm = TRUE),
      median_abundance_change = median(diffabundance, na.rm = TRUE),
      mean_temporal_shift = mean(diffyears, na.r=TRUE),
      n_populations = n()
    ) %>%
    ungroup()

  write.csv(summary_df_species_simple, file = paste0("../dataLPI/summary_df_species_simple.csv"))

} else {
  # Read precomputed summary file
  summary_df_species_simple <- read.csv(file = paste0("../dataLPI/summary_df_species_simple.csv"))
}

# Get simple statistic
meanAx <- summary_df_species_simple %>%
  dplyr::filter(mean_abundance_change < 0) %>%
  pull(mean_abundance_change)

meanAx <- -meanAx

Xpi <- (1 - (1 - meanAx)^0.05)

```

As a though experiment -- and to avoid controversial discussion on how or whether to measure abundance changes from LPI data -- I simply take the earliest and latest census value for each LPI tracked population and compute a arithmetic average if a species has multiple observations. This yields -`r printmean(meanAx*100)`% [IQR = -`r printIQR(meanAx*100)`%]. The median number of populations is: `r printmedian(summary_df_species_simple$n_populations)` populations [IQR = `r printIQR(summary_df_species_simple$n_populations)`]. The median number of years in between earliest and latest observation is: `r printmedian(summary_df_species_simple$mean_temporal_shift)` years [IQR = `r printIQR(summary_df_species_simple$mean_temporal_shift)` years].

A thought experiment back-calculation of genetic diversity loss can be created using, for instance, scenario 2a, $X_{\pi}=$ `r printmean(Xpi*100)`% [IQR `r printIQR(Xpi*100)`%].

\clearpage

## Text S7: Limitations, caveats, and thoughts forward

### Thoughts on data limitations

-   Published studies may have too short timelines to detect signals. About 10 years or 2 generations on average are in evolutionary trends effectively instantaneous and even if populations are undergoing bottlenecks and drift changes are expected to be small. Possibly applying historical/ancient DNA technologies would enable the temporal breadth backwards (see also next point). This will help avoiding the "baseline shifting problem" in biodiversity, where our reference point of what "original diversity" was shifts constantly leading to having a recent past (already eroded nature) as our pristine nature reference.

-   A problem in genomic monitoring that we are facing now and will continue happening in the future is that genetic technologies are moving fast. Therefore, comparing observations published in the literature 10 years ago is today considered obsolete genetic technologies, so meta-analysis will not make use of the best current data. This is happening in the Shaw dataset, a typical molecular ecology study measuring genetic composition of a population may include 10 microsatellites for genotyping. Nowadays, even moderate budget projects could possibly use genomic technologies to generate much more accurate values of genetic diversity (NB the measurement error rate scales 1/number of bp sequenced). This can be done even using pool sequencing (i.e. sequencing multiple individuals' DNA in one tube) since one aims to get a population-level diversity estimate. In addition, while many genomic approaches need of assembling a genome blueprint for a species, Kmer-based approaches (digesting sequencing reads directly in DNA sequence tables) has proven to be efficient at characterizing genetic diversity (Roberts & Josephs et al. 2025, Evolution Letters).

-   A problem in genetic data often ignored or not reported is the scale of measurement of genetic diversity. While population genetics classically describes the individual, the population, and the species-which has been adopted to essential biodiversity variables (Hoban et al. 2022 Biological Reviews)-these three levels are not well captured in sampling protocols nor in policy texts (CBD). The difficulty is that genetic diversity within a population may change while it not changing at the species-level scale. Likewise genetic diversity within an individual (heterozygosity or inbreeding coefficient) may change but diversity at the population is maintained. Different demographic processes, evolutionary processes, may affect these without them being problematic, in fact they may go in different directions. It seems all levels are relevant (intuitively perhaps within-species may be the most relevant) and analyses should be conducted at all such three scales.

### Thoughts on inverse model limitations

Apart from data limitations, the inverse modeling proposed in the text also has limitations and possible improvements. No model is going to be perfect and one needs to balance usability (e.g. back of the envelope predictions vs complex process-based simulations). Here I mostly opted for back of the envelope mathematical predictions such as single population dynamics or genetic diversity-area relationships as these are very powerful to scale to many species (scenario 1a,1b,2a). For more accurate predictions and dynamics I also included some simulation/numeric based approaches (scenario 2b, Mualim et al 2024 bioRxiv) which can be extended to any population geographic conformation wanted, starting population size, migration rates, but these require further information per species.

-   Shifting baseline problem underestimates impacts: Many species have already decreased in population sizes and thus contemporary measurements used in modeling (bottlenecks - to - genetics) or inverse modeling (genetic - to - bottlenecks) equations will underestimate past declines of genetic diversity before either genetic monitoring or popualtion monitoring were established.

-   Survival bias underestimates impacts: by definition genetic monitoring is only going to capture genetic diversity within a population, but cannot quantify genetic diversity loss of unsampled or extinct populations. It is likely that many populations within species are being altered with habitat losses (Exposito-Alonso 2022 Science) and thus genetic diversity loss should be larger than, e.g., what is inferred in Fig. 2.

-   Wild populations often do not follow Wright-Fisher evolutionary assumptions, modeling may under- or over-estimate impacts. The inferences based on evolutionary theory equations, equilibrium, sampling, necessarily use assumptions. Some assumptions involve no natural selection, random outcrossing of individuals (scenario 1a/1b), equal reproduction, etc. Underestimation may occur if, for instance, bottlenecks are impacting populations' Ne to a larger degree due to selfing or other. Models may overestimate impacts of bottlenecks, for instance, when in nature populations have overlapping generations that maintain genetic diversity.

-   Unknown refugia/reservoirs of populations containing much of a species genetic diversity will protect it to impacts elewhere thus models may over-estimate genetic diversity loss.

```{=html}
<!-- -->
```
-   Historical trends have legacy trajectories in genetic diversity. Species are not static, and are often beem changing in population size for long periods of time. Whether these are upward or downward, will have effects in genetic diversity trajectories in the future (even with contemporary rapid changes in population).
